<!DOCTYPE html>
<html>

	<head> 
     <meta charset="utf-8">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
   	 </head> 

	<div id = "title">
	  <p>
	  		<p><img src="{{ site.baseurl }}/images/Geno_Diver_Logo_Small.png" ></p>
	  </p>
	</div>
	
	
	<div id = "parameterclass">
	 	<font color="black"> Simulating a Fitness Trait </font>
	</div>	 	
	<br> <br>

	<div id = "textbox">
		<div id = "text">
	 		&emsp; The parameter file below illustrates how to simulate a fitness trait only. When simulating a fitness trait it is important to 
	 		ensure that you have enough founder individuals because a portion will not make it to breeding age. Therefore, an extra 100 males and 
	 		200 females were added to the founder population to ensure enough individuals are available to generate the breeding population. This 
	 		will also impact the replacement rate because if enough progeny aren't available to remain at the chosen male and female population size 
	 		the simulation will exit. A full description of how the fitness value of an individual impacts its ability to make it to breeding age 
	 		is described in the "QTL/FTL Distribution Parameters" link.
	 		<br> <br>	  
		</div>
	</div>
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			−−−−−−−|    Simulating a Fitness Trait   |−−−−−−− <br>
			−| General |− <br>
			START: sequence <br>
			SEED: 1500 <br>
			−| Genome & Marker |− <br>
			CHR: 3 <br>
			CHR_LENGTH: 150 150 150 <br>
			NUM_MARK: 4000 4000 4000 <br>
			QTL: 0 0 0 <br>
			FIT_LETHAL: 20 20 20  <br>
			FIT_SUBLETHAL: 200 200 200 <br>
			−| Population |− <br>
			FOUNDER_Effective_Size: 250 <br>
			MALE_FEMALE_FOUNDER: 150 600 random 0 <br>
			VARIANCE_A: 0.0 <br>
			VARIANCE_D: 0.0 <br>
			−| Selection |− <br>
			GENERATIONS: 25 <br>
			INDIVIDUALS: 50 0.2 400 0.2 <br>
			PROGENY: 1 <br>
			SELECTION: random high <br>
			CULLING: random 10 <br>
			-| Mating |- <br>
			MATING: random <br>
			-| OUTPUT OPTIONS |- <br>
			GENOTYPES: no <br>
		</div>
	</div>
	
	<br> <br>

	<div id = "parametersummary">
		<b> Parameter File Summary </b> <br>
		<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of 
			short-range LD (250). The SNP panel contains 12,000 marker (i.e. 4,000 markers per chromosome). For each chromosome, 20 lethal and 200 
			sub-lethal mutations were generated. The quantitative trait has a broad sense heritability of 0.0 and therefore an animals phenotype is 
			only a function of random environmental deviations with a variance of 1.0. The founder population consisted of 150 males and 600 females. 
			For each generation, a total of 50 males and 400 females are in the population. A total of 10 and 80 (0.2 replacement rate) male and female 
			parents, respectively, are culled and replaced by new progeny each generation. Starting at the first generation animals are randomly selected 
			or culled each generation. The maximum number of generations an animal can remain in the breeding population is 10. Each mating pair produced 
			one progeny and parents were mated at random. <br>
		 </div>
	</div>
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; Inspection of the log file will provide summary statistics on the mean selection coefficients and degree of dominance for the lethal and sub-lethal 
			fitness effect along with their associated frequency. The default settings for the lethal mutations result in a high selection coefficient (0.72) and very little 
			dominance (0.001). Therefore the heterozygote is normal, while the unfavorable homozygote has a low fitness value. The settings for the sub-lethal mutations result in a 
			lower selection coefficient (0.03) and moderate degree of dominance (0.30). Therefore the heterozygote now has a reduced fitness value compared to the fittest homozygote. 
			Lastly, the mean allele frequency of the unfavorable allele is higher for sub-lethal compared to lethals. Below is a screenshot of the lines in the log file that display 
			the summary statistics.
		</div>
	</div>
	
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example4a.png" ></p>
	</div>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; When a fitness trait is simulated, the animals that died due to fitness are placed in the "Low_Fitness" along with summary statistics for the respective animals. 
			 Utilizing the R code outlined below the following plots were generated to illustrate the type of information in the "Low_Fitness" file. <br>
		</div>
	</div>
	
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2); require(gridExtra); require(grid) <br>
			setwd("/Users/jeremyhoward/Desktop/C++Code/SamplingGamates/17_GenoDiver_V2/GenoDiverFiles") ## Change <br>
			############################################################################## <br>
			### Correlation between fitness and number of homozygous lethal mutations  ### <br>
			############################################################################## <br>
			## Animals that Lived ## <br>
			df <- read.table(file="Master_DataFrame",sep=" ",header=TRUE) <br>
			ggplot(df, aes(x=Fitness, y =as.factor(Homolethal))) + geom_point(colour = "red", size = 3) +  <br>
			  labs(title = "Animals that Lived", x = "Fitness", y = "Number of Homozygous Lethal") + theme_bw() + <br>
			  theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12),axis.text=element_text(size=10)) <br>
			## Animals that Died ## <br>
			df <- read.table(file="Low_Fitness",sep=" ",header=TRUE) <br>
			ggplot(df, aes(x=Fitness, y = as.factor(Homolethal))) + geom_point(colour = "red", size = 3) + <br>
 			&emsp; labs(title = "Animals that Died", x = "Fitness", y = "Number of Homozygous Lethal") + theme_bw() + <br>
			&emsp; theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12),axis.text=element_text(size=10)) <br><br>
			############################################################## <br>
			### Grab All Unique Sires and Plot Number of deaths by Sire ## <br>
			############################################################## <br>
			df <- read.table(file="Low_Fitness",sep=" ",header=TRUE,colClasses = c(rep("numeric",16),"character")) <br>
			## Get death counts by sire ## <br>
			SireDeath <- aggregate(GV ~ Sire, data=df,FUN=length) # get count by sire <br>
			SireDeath <- SireDeath[which(SireDeath$Sire != 0), ] # remove unknown sire groups <br>
			SireDeath <- SireDeath[order(-SireDeath$GV), ] # order largest to smallest <br>
			ggplot(data=SireDeath, aes(SireDeath$GV)) + geom_histogram() + theme_bw() + <br>
			&emsp; ylab ("Count of Sires by Number of Dead Progeny ") + xlab ("Number of Deaths for a Sire") + <br>
			&emsp; ggtitle("Histogram of Number of \n Progeny that Died by Sire") + theme_bw() + theme(plot.title = element_text(hjust = 0.5)) <br>
		</div>
	</div>
	<br>
		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example4Figures.png" ></p>
	</div>
				
	<div id = "homepage">
		<a href="{{site.baseurl}}/index.html">Home Page</a>
	</div>
	
	
	

</html>

