<!DOCTYPE html>
<html>

	<head> 
     <meta charset="utf-8">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
   	 </head> 

	<div id = "title">
	  <p>
	  		<p><img src="{{ site.baseurl }}/images/Geno_Diver_Logo_Small.png" ></p>
	  </p>
	</div>
	
	
	<div id = "parameterclass">
	 	<font color="black"> Simulating a Fitness Trait </font>
	</div>	 	
	<br> <br>
	
	GOT HERE!!
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; Similar to the previous example, a bash script is outlined below that illustrates how to reduce the memory requirements by truncating
	 		animals born 'n' generations ago. This is especially useful if you want to simulate a large number of generations (25+) and as outlined below
	 		truncating the data to only include recently born animals does not have a large impact on the genetic trend across generations. To truncate animals for BLUP 
	 		based ebv prediction the "BLUP_OPTIONS" parameter needs to be included and the first option specifies how many generations back from the current 
	 		group of selection candidates you want to go. For example, if a value of 1 is specified the selection candidate’s parents and the associated parents 
	 		progeny will only be included in the analysis. The default value is the number of generations simulated (i.e. all animals are used each generation). 
	 		The bash script outlined below uses the 'gblup' option to generate breeding values but either uses all the animals, 3 or 1 generations back from the 
	 		selection candidates to estimate breeding values. A total of 10 replicates were generated. After a scenario is done the directory where the replicates 
	 		are saved is renamed within the bash script.
	 		<br> <br>	  
		</div>
	</div>
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			##-------------------------------------------------------------------------------- <br>
			## First generate parameter file. Put parameters to change <br>
			## at the very bottom of the file. That way you can use the <br>
			## 'head' linux command for the parameters that change and <br>
			## use the 'echo' linux command to add any new parameters. <br>
			##-------------------------------------------------------------------------------- <br>
			## In case 5_Example5.txt is still a file ## <br>
			rm -rf 5_Example5.txt || TRUE <br>
			<br>
			# Parameters that are the same # <br>
			echo '-| General |-' >> 5_Example5.txt <br>
			echo 'SEED: 1500' >> 5_Example5.txt <br>
			echo 'NREP: 10' >> 5_Example5.txt <br>
			echo '−| Genome & Marker |−' >> 5_Example5.txt <br>
			echo 'CHR: 3' >> 5_Example5.txt <br>
			echo 'CHR_LENGTH: 150 150 150' >> 5_Example5.txt <br>
			echo 'NUM_MARK: 4000 4000 4000' >> 5_Example5.txt <br>
			echo 'QTL: 50 50 50' >> 5_Example5.txt <br>
			echo '−| Population |−' >> 5_Example5.txt <br>
			echo 'FOUNDER_Effective_Size: Ne100_Scen1' >> 5_Example5.txt <br>
			echo 'MALE_FEMALE_FOUNDER: 50 400 random 5' >> 5_Example5.txt <br>
			echo 'VARIANCE_A: 0.45' >> 5_Example5.txt <br>
			echo 'VARIANCE_D: 0.05' >> 5_Example5.txt <br>
			echo '−| Selection |−' >> 5_Example5.txt <br>
			echo 'GENERATIONS: 20' >> 5_Example5.txt <br>
			echo 'INDIVIDUALS: 50 0.2 400 0.2' >> 5_Example5.txt <br>
			echo 'PROGENY: 1' >> 5_Example5.txt <br>
			echo 'SELECTION: ebv high' >> 5_Example5.txt <br>
			echo 'CULLING: ebv 8' >> 5_Example5.txt <br>
			echo 'MATING: random125 simu_anneal' >> 5_Example5.txt <br>
			echo 'EBV_METHOD: gblup' >> 5_Example5.txt <br>
 			echo 'G_OPTIONS: VanRaden observed' >> 5_Example5.txt <br>
			<br>
			##-------------------------------------------------------------------------------- <br>
			## First do GBLUP with all animals ## <br>
			##-------------------------------------------------------------------------------- <br>
			echo 'START: sequence' >> 5_Example5.txt <br>
			echo 'BLUP_OPTIONS: 20 pcg cholesky' >> 5_Example5.txt <br>
			## Run GenoDiver ## <br>
			./GenoDiver 5_Example5.txt <br>
			## rename replicate output to reps_20trunc ## <br>
			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_20trunc <br>
			<br>
			##-------------------------------------------------------------------------------- <br>
			## Then do GBLUP with 3 generations back  ## <br>
			##-------------------------------------------------------------------------------- <br>
			## Remove parameters that are changing ## <br>
			head -n 22 5_Example5.txt > 5_Example5a.txt <br>
			mv ./5_Example5a.txt ./5_Example5.txt <br>
			## echo in new paramters ## <br>
			echo 'START: founder' >> 5_Example5.txt <br>
			echo 'BLUP_OPTIONS: 3 pcg cholesky' >> 5_Example5.txt <br>
			## Run GenoDiver ## <br>
			./GenoDiver 5_Example5.txt <br>
			## rename replicate output to reps_3trunc ## <br>
 			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_3trunc <br>
			<br>
			##-------------------------------------------------------------------------------- <br>
			## Then do GBLUP with 1 generation back  ## <br>
			##-------------------------------------------------------------------------------- <br>
			## Remove parameters that are changing ## <br>
			head -n 22 5_Example5.txt > 5_Example5a.txt <br>
			mv ./5_Example5a.txt ./5_Example5.txt <br>
			## echo in new parameters <br>
			echo 'START: founder' >> 5_Example5.txt <br>
			echo 'BLUP_OPTIONS: 1 pcg cholesky' >> 5_Example5.txt <br>
			## Run GenoDiver ## <br>
			./GenoDiver 5_Example5.txt <br>
			## rename replicate output to reps_1trunc ## <br>
			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_1trunc <br>
		</div>
	</div>
	
	<br> <br>
	
	<div id = "parametersummary">
		<b> Parameter File Summary </b> <br>
		<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of 
			short-range LD (Ne100_Scen1). The SNP panel contains 12,000 marker (i.e. 4,000 markers per chromosome). For each chromosome, 150 randomly 
			placed QTL were generated, and no FTL mutations were generated. The narrow and broad sense heritability for the trait is 0.45 and 0.50, 
			respectively. The phenotypic variance is by default set at 1.0, and therefore the residual variance is 0.5. The founder population consisted 
			of 50 males and 400 females. For each generation, a total of 50 males and 400 females are in the population. A total of 10 and 80 (0.2 
			replacement rate) male and female parents, respectively, are culled and replaced by new progeny each generation. Random selecton of progeny 
			and culling of parents was conducted for 5 generations. After 5 generations, animals with a high EBV were selected or culled each generation. 
			The EBV are estimated using a genomic-based BLUP utilizing all the animals or truncating the dataset 3 and 1 generations back from the 
			selection candidates. The genomic relationship matrix was constructed based on the observed frequencies for the animals utilized to calculate 
			breeding values. Each mating pair produced one progeny. Parents that had pedigree-based relationships greater than 0.125 were avoided, 
			and this was optimized based on the simulated annealing method. <br>
		 </div>
	</div>
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; Similar to Example 4, the important files are saved within the renamed replicate folder within each scenario. The replicate number is appended to the file name.
			The bash script above renames the replicate directories to "reps_20trunc", "reps_3trunc" and "reps_1trunc" for the model that uses all animals, 3 generations back or 
			1 generation back, respectively. <br> <br>
			&emsp; Utilizing the R code outlined below the following plot was generated to illustrate how to loop through each scenario and generate plots
			that describe the impact of truncating animals on the genetic trend and the reduction in the number of equations to solve. <br>
		</div>
	</div>
	
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2) <br>
			wd <- "/Users/jeremyhoward/Desktop/" ## Change <br> <br>
			## the directory name for each scenario ## <br>
			scen <- c("reps_20trunc","reps_3trunc","reps_1trunc") <br>
			## Number of replicates simulated ## <br>
			reps <- c(1500:1509) <br> <br>
			################################################################## <br>
			## Loop through and grab metric across scenarios and replicates ## <br>
 			################################################################## <br>
			for(i in 1:length(scen)) <br>
			{ <br>
			&emsp; for(j in 1:length(reps)) <br>
  			&emsp; { <br>
    		&emsp;&emsp; filename <- paste(wd,scen[i],"/Summary_Statistics_DataFrame_Performance_",reps[j],sep="") <br>
    		&emsp;&emsp; df <- read.table(file=filename,header=TRUE,colClasses=c("numeric",rep("character",6))) # read in file <br>
    		&emsp;&emsp; df$Method <- unlist(strsplit(scen[i],"_"))[2]  # save method <br>
    		&emsp;&emsp; df$Rep <- reps[j];  # save replicate <br>
    		&emsp;&emsp; df$Value <- matrix(unlist(strsplit(df[, 4],"\\(")),ncol=2,byrow=TRUE)[,1] # grab mean gv by generation <br>
    		&emsp;&emsp; if(j == 1 & i == 1){summarytable <- df[ ,c(8,9,1,10)]} <br>
    		&emsp;&emsp; if(j > 1 | i > 1){summarytable <- rbind(summarytable,df[ ,c(8,9,1,10)]);} <br>
  			&emsp; } <br>
			} <br>
			## generate mean and sd by generation and method <br>
			summarytable$Value <- as.numeric(paste(summarytable$Value)) <br>
			means <- aggregate(Value ~ Generation + Method, data=summarytable,FUN=mean) <br>
			sds <- aggregate(Value ~ Generation + Method, data=summarytable,FUN=sd) <br> <br>
			################################################# <br>
			## Plot Genetic Trend across Different Methods ## <br>
			################################################# <br>
			plotdf <- cbind(means,sds[,3]); rm(means,sds) <br>
			names(plotdf) <- c("Generation","Method","Mean","SD") <br> 
			pd <- position_dodge(0.5) <br><br>
			ggplot(plotdf, aes(x=Generation, y=Mean, colour=Method)) +  <br>
  			&emsp; geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD), colour="black", width=.4, size = 0.5, position=pd) + <br>
  			&emsp; geom_point(size=2.0) + geom_line(size=0.50) + theme_bw() + <br>
  			&emsp; labs(title = "Genetic Trend (95% CI)", x = "Generation", y = "Mean True Breeding Value") + <br>
  			&emsp; theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12), <br>
        	&emsp; legend.position="bottom",axis.text=element_text(size=10)) <br> <br>
			############################################################################## <br>
			## Loop through and grab size of LHS within log_file.txt for each replicate ## <br>
			############################################################################## <br>
			for(i in 1:length(scen)) <br>
			{ <br>
			&emsp; for(j in 1:length(reps)) <br>
  			&emsp; { <br>
    		&emsp;&emsp; filename <- paste(wd,scen[i],"/log_file_",reps[j],sep="") <br>
    		&emsp;&emsp; df <- scan(file=filename,sep="\n",what=character(),quiet = TRUE);  ## save each line as a string ## <br>
    		&emsp;&emsp; ## Grab generation ## <br>
    		&emsp;&emsp; tempa <- df[grep("Begin Generation",df)]; tempa <- tempa[6:length(tempa)-1] <br>
    		&emsp;&emsp; tempa <- as.numeric(matrix(unlist(strsplit(tempa," ")),ncol = 5,byrow=TRUE)[ ,c(4)]) <br>
    		&emsp;&emsp; ## Grab size of Relationship Matrix and add one for intercept ## <br>
    		&emsp;&emsp; tempb <- df[grep("Size of Relationship Matrix",df)]; tempb <- unlist(strsplit(tempb," ")); tempb <- tempb[which(tempb != "")] <br>
    		&emsp;&emsp; tempb <- matrix(tempb,ncol = 8,byrow=TRUE); tempb <- as.numeric(tempb[c(1:(nrow(tempb)-1)),c(6)]);  tempb <- tempb+1; <br>
    		&emsp;&emsp; ## combine together and add variables <br>
    		&emsp;&emsp; df <- data.frame(cbind(tempa,tempb),stringsAsFactors = FALSE) <br>
    		&emsp;&emsp; df$Method <- unlist(strsplit(scen[i],"_"))[2]  # save method <br>
    		&emsp;&emsp; df$Rep <- reps[j];  # save replicate <br>
    		&emsp;&emsp; names(df) <- c("Generation","LHS_Size","Method","Rep") <br>
    		&emsp;&emsp; if(j == 1 & i == 1){summarytable <- df;} <br>
    		&emsp;&emsp; if(j > 1 | i > 1){summarytable <- rbind(summarytable,df);} <br>
  			&emsp;} <br>
			} <br>
			## generate mean and sd by generation and method <br>
			summarytable$LHS_Size <- as.numeric(paste(summarytable$LHS_Size)) <br>
			means <- aggregate(LHS_Size ~ Generation + Method, data=summarytable,FUN=mean) <br>
			sds <- aggregate(LHS_Size ~ Generation + Method, data=summarytable,FUN=sd) <br> <br>
			################################################# <br>
			## Plot LHS Size For Different Truncation Methods ## <br>
			################################################# <br>
			plotdf <- cbind(means,sds[,3]); rm(means,sds) <br>
			names(plotdf) <- c("Generation","Method","Mean","SD") <br>
			pd <- position_dodge(0.5) <br> <br>
			ggplot(plotdf, aes(x=Generation, y=Mean, colour=Method)) + <br>
  			&emsp; geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD), colour="black", width=.4, size = 0.5, position=pd) + <br>
  			&emsp; geom_point(size=2.0) + geom_line(size=0.50) + theme_bw() + <br>
  			&emsp; labs(title = "Reduction in Number of Equations", x = "Generation", y = "Left Hand Side Size") + <br>
  			&emsp; theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12), <br>
        	&emsp; legend.position="bottom",axis.text=element_text(size=10)) <br>
		</div>
	</div>
	<br>
		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example5Figures.png" ></p>
	</div>
				
	<div id = "homepage">
		<a href="{{site.baseurl}}/index.html">Home Page</a>
	</div>
	
	
	

</html>

