<!DOCTYPE html>
<html>

	<head> 
     <meta charset="utf-8">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
   	 </head> 

	<div id = "title">
	  <p>
	  		<p><img src="{{ site.baseurl }}/images/Geno_Diver_Logo_Small.png" ></p>
	  </p>
	</div>
	
	
	<div id = "parameterclass">
	 	<font color="black"> Multiple Progeny Quantitative Trait </font>
	</div>	 	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; The parameter file outlined below illustrates how to simulate a prolific species (i.e. Swine) with a 
	 		threshold on the minimum number of progeny to select within each family. <br> <br>	  
		</div>

	</div>

	<div id = "raisedbox">
		<div id = "raisedboxtext">
			−−−−−−−|     Multiple Progeny Quantitative Trait   |−−−−−−− <br>
			−| General |− <br>
			START: sequence <br>
			SEED: 1501 <br>
			−| Genome & Marker |− <br>
			CHR: 3 <br>
			CHR_LENGTH: 150 150 150 <br>
			NUM_MARK: 4000 4000 4000 <br>
			QTL: 50 50 50 <br>
			−| Population |− <br>
			FOUNDER_Effective_Size: Ne70 <br>
			MALE_FEMALE_FOUNDER: 25 200 random 0 <br>
			VARIANCE_A: 0.35 <br>
			VARIANCE_D: 0.05 <br>
			−| Selection |− <br>
			GENERATIONS: 10 <br>
			INDIVIDUALS: 25 0.2 200 0.2 <br>
			PROGENY: 8 <br>
			MAXFULLSIB: 2 <br>
			SELECTION: ebv high <br>
			EBV_METHOD: pblup <br>
			CULLING: ebv 5 <br>
			-| Mating |- <br> 
			MATING: random125 simu_anneal <br>
		</div>
	</div>
	
	<br> <br>
	
	<div id = "parametersummary">
		<b> Parameter File Summary </b> <br>
		<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of short-range LD 
			(Ne70). The SNP panel contains 12,000 marker (i.e. 4,000 markers per chromosome). For each chromosome, 50 randomly placed QTL were generated, and no FTL 
			mutations were generated. The narrow and broad sense heritability for the trait is 0.10 and 0.105, respectively. The phenotypic variance is by default set 
			at 1.0, and therefore the residual variance is 0.895. The founder population consisted of 25 males and 200 females. For each generation, a total of 25 males 
			and 200 females are in the population. A total of 5 and 40 (0.2 replacement rate) male and female parents, respectively, are culled and replaced by new 
			progeny each generation. A total of 10 generations were simulated. Animals with a high estimated breeding value (EBV) were selected or culled each 
			generation. The EBV are estimated using an animal model based on a pedigree-based relationship matrix. Each mating pair produced eight progeny. Within each
			full-sib family a maximum of 2 progeny can be selected. Parents that had pedigree-based relationships greater than 0.125 were avoided, and 
			this was optimized based on the simulated annealing method. <br>
		 </div>
	</div>
	<br>
	<div id = "textbox">
		<div id = "text">
			&emsp; Once the program has finished, inspection of the log file will provide details on the impact of the "MAXFULLSIB" option. Within the log file, 
			after the "Begin ebv Selection of offspring" section, the number of times "n" number of siblings were selected within a family is outlined for each 
			generation. This provides an overview of the degree of co-selection of siblings that occur based on the simulation design outlined above. If this option 
			would not have been included, the maximum number of siblings that can be selected within a family would be 8. Given the low narrow-sense heritability 
			of the trait, the number of siblings selected within a family would be higher if this parameter wasn't included.
		</div>
	</div>
	
	<div id = "homepage">
		<a href="{{site.baseurl}}/index.html">Home Page</a>
	</div>
	
	
	

</html>

