<!DOCTYPE html>
<html>

	<head> 
     <meta charset="utf-8">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
   	 </head> 

	<div id = "title">
	  <p>
	  		<p><img src="{{ site.baseurl }}/images/Geno_Diver_Logo_Small.png" ></p>
	  </p>
	</div>
	
	
	<div id = "parameterclass">
	 	<font color="black"> Overview of Summary Statistics. </font>
	</div>	 	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; A large number of summary statistics that range from genetic diversity, genetic values and fitness related metrics are generated by default within Geno-Diver 
	 		for each generation. A subset of these summary statistics have been outlined in the previous examples and each one is described in detail within the "Output Files" 
	 		link on the homepage. The user can output even more summary statistics by including extra parameters in a parameter file that are by default not calculated. The 
	 		supplemental summary statistics include linkage dis-equilibrium decay, linkage phase persistance, run-of-homozygosity across the genome and the true additive and dominance 
	 		variation across the genome for each generation. The parameter file below illustrates the parameters to include to generate the previously mentioned summary 
	 		statistics.
	 		<br> <br>	  
		</div>
	</div>
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			−−−−−−−|     Overview of Summary Statistics   |−−−−−−− <br>
			−| General |−  <br>
			START: sequence <br>
			SEED: 1501 <br>
			−| Genome & Marker |− <br>
			CHR: 3 <br>
			CHR_LENGTH: 150 150 150 <br>
			NUM_MARK: 4000 4000 4000 <br>
			QTL: 200 200 200 <br>
			−| Population |− <br>
			FOUNDER_Effective_Size: Ne100_Scen1 <br>
			MALE_FEMALE_FOUNDER: 50 400 random 0 <br>
			VARIANCE_A: 0.25 <br>
			VARIANCE_D: 0.05 <br>
			−| Selection |− <br>
			GENERATIONS: 20 <br>
			INDIVIDUALS: 50 0.2 400 0.2 <br>
			PROGENY: 1 <br>
			SELECTION: ebv high <br>
			PARITY_MATES_DIST: 2.0 1.0 
			EBV_METHOD: pblup <br>
			CULLING: ebv 10 <br>
			-| Mating |- <br>
			MATING: random <br>
			-| OUTPUT OPTIONS |- <br>
			OUTPUT_LD: yes <br>
			GENOME_ROH: 5 5 10 15 20 <br>
			WINDOWQTLVAR: yes <br>
		</div>
	</div>
	
	<br> <br>
	
	<div id = "parametersummary">
		<b> Parameter File Summary </b> <br>
		<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of short-range LD 
			(Ne100_Scen1). The SNP panel contains 12,000 marker (i.e. 4,000 markers per chromosome). For each chromosome, 200 randomly placed QTL were generated, and no FTL 
			mutations were generated. The narrow and broad sense heritability for the trait is 0.25 and 0.30, respectively. The phenotypic variance is by default set 
			at 1.0, and therefore the residual variance is 0.7. The founder population consisted of 50 males and 400 females. For each generation, a total of 50 males 
			and 400 females are in the population. A total of 10 and 80 (0.2 replacement rate) male and female parents, respectively, are culled and replaced by new 
			progeny each generation. A total of 20 generations were simulated. Animals with a high estimated breeding value (EBV) were selected or culled each 
			generation. The EBV are estimated using an animal model based on a pedigree-based relationship matrix. Each mating pair produced one progeny and 
			parents were mated at random. The mating distribution was skewed so that older animals had more mating pairs than younger animals. <br>
		 </div>
	</div>
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; Outlined below is a summary of each metric and some R code that generates a plot of the data.
		</div>
	</div>
	<br> <br>
	 
	<div id = "parameterclass">
	 	<font color="black"> LD Decay Across Generations </font>
	</div>
	<br>
	<div id = "textbox">
		<div id = "text">
			&emsp; Within the "LD_Decay" file the average correlation (r2) between two SNP across a range of distances is outlined. The first row specifies the distance in 
			kilobases. Any row after the first is then the LD decay for a given generation. The code below grabs the LD decay for generations 0, 5, 10, 15 and 20 and plots 
			the LD decay by generation. When specifying your own founder population history this can be utilized to determine whether you have the LD decay you desire.
		</div>
	</div>
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2); require(gridExtra); require(grid) <br>
			setwd("/Users/jeremyhoward/Desktop/C++Code/SamplingGamates/17_GenoDiver_V2/GenoDiverFiles") ## Change <br>
			######################### <br>
			### Read in LD_Decay file  ### <br>
			######################### <br>
			df <- read.table(file="LD_Decay",sep=" ",header=FALSE); <br>
			Distance <- df[1, ]; df <- df[-c(1), ]; # grab distance <br>
			# Plot for generations 0 5 10 15 20 <br>
			generations <- c(1,6,11,16,21) <br>
			## Make dataframe for ggplot2 <br>
			for(i in 1:length(generations)) <br>
			{ <br>
			&emsp; tempdf <- data.frame(cbind(t(Distance),t(df[generations[i], ])),stringsAsFactors = FALSE) <br>
  			&emsp; tempdf$gen <- generations[i] -1; <br>
 			&emsp; names(tempdf) <- c("Distance","r2","Generation") <br>
  			&emsp; if(i == 1){plotdf <- tempdf;rm(tempdf)} <br>
  			&emsp; if(i > 1){plotdf <- rbind(plotdf,tempdf); rm(tempdf)} <br>
			} <br>
			plotdf$Generation <- as.factor(as.numeric(plotdf$Generation)) <br>
			ggplot(plotdf, aes(x=Distance, y = r2, colour = Generation)) + geom_point(size = 1) + <br> 
  			&emsp; labs(title = "LD Decay", x = "Distance (kb)", y = "Correlation between two SNP") + theme_bw() + <br>
  			&emsp; theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12),legend.position="bottom",axis.text=element_text(size=10)) <br>
  		</div>
	</div>		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example9aFigures.png" ></p>
	</div>
	
	<div id = "parameterclass">
	 	<font color="black"> LD Decay around QTL </font>
	</div>
	<br>
	<div id = "textbox">
		<div id = "text">
			&emsp; Within the "QTL_LD_Decay" file the average correlation (r2) between a SNP within 2.5 Mb of a QTL and the associated QTL is outlined across all QTL. The 
			first column represents the chromosome and the second column represents the position in Mb. The average correlation was binned into distances of 0 - 0.5 Mb, 
			0.5 - 1.0 Mb, 1.0 - 1.5 Mb, 1.5 - 2.0 Mb and 2.0 - 2.5 Mb from the QTL. The last column is the average correlation within a generation across the different 
			window sizes. Correlations within a generation across windows are separated by a ":" and sets of correlations across generations are seperated by a "_". 
			The R code below outlines the change in LD decay surrounding QTL with a large change in allele frequency across generations. The change in LD is an average across all
			5 window categories.
		</div>
	</div>
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2); require(gridExtra); require(grid) <br>
			setwd("/Users/jeremyhoward/Desktop/C++Code/SamplingGamates/17_GenoDiver_V2/GenoDiverFiles") ## Change <br>
			######################### <br>
			### LD Decay around QTL  ### <br>
			######################### <br>
			df <- read.table(file="QTL_LD_Decay",sep=" ",header=TRUE,colClasses = c("numeric","numeric","character")); ## Read in file ## <br>
			qtl <- read.table(file="QTL_new_old_Class",sep=" ",header=TRUE,colClasses = c("numeric",rep("character",6))) ## Read in QTL files ## <br>
			## Find ones that changed a lot from low to high and calculate mean LD across windows by generation by generation for each one  ## <br>
			freq <- matrix(unlist(strsplit(qtl[,7],"_")),nrow=nrow(qtl),byrow=TRUE) <br>
			freq <- apply(freq,2,as.numeric) <br>
			X <- which(abs((freq[,1] - freq[,21])) > 0.50) <br>
			for(i in 1:length(X)) <br>
			{ <br>
  			&emsp; # stringsplit first by generation than by window # <br>
  			&emsp; temp <- apply(matrix(unlist(strsplit(unlist(strsplit(df[X[i],3],"_")),":")),ncol=5,byrow=TRUE),2,as.numeric) <br>
  			&emsp; if(as.numeric(length(which(temp == -5))) == 0)  # make sure all were able to be estimated # <br>
  			&emsp; { <br>
  			&emsp;&emsp; temp <- rowMeans(temp) <br>
  			&emsp;&emsp; temp <- data.frame(cbind(rep(i,length(temp)),temp,c(1:length(temp)))) <br>
  			&emsp;&emsp; if(i == 1){plotdf <- temp;rm(temp)} <br>
  			&emsp;&emsp; if(i > 1){plotdf <- rbind(plotdf,temp); rm(temp)} <br>
  			&emsp; } <br>
			} <br>
			names(plotdf) <- c("QTL","r2","Generation") <br>
			plotdf$QTL <- as.factor(plotdf$QTL) <br>
			ggplot(plotdf, aes(x=Generation, y = r2, colour = QTL)) + geom_line(size = 1) + <br>
			&emsp; labs(title = "Change in LD Around QTL \n with Large Allele Freqency Changes", x = "Generation", y = "Mean correlation between SNP and QTL") + theme_bw() + <br>
			&emsp; theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12),legend.position="none",axis.text=element_text(size=10)) <br>
  		</div>
	</div>		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example9bFigures.png" ></p>
	</div>
	
	<div id = "parameterclass">
	 	<font color="black"> Persistence of Phase </font>
	</div>
	<br>
	<div id = "textbox">
		<div id = "text">
			&emsp; Within the "Phase_Persistance_Generation" file the correlation between LD phase within different window sizes across generations is estimated. The LD phase is the
			square root of the correlation between a SNP and QTL and is the same sign as the LD parameter (D). he first column is the generation and the columns following would be 
			the correlation between the current generation and preceding generations. The R code below outlines the change in phase persistence for the animals born in generation 20 
			compared to the preceding generations by window size.
		</div>
	</div>
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2); require(gridExtra); require(grid) <br>
			setwd("/Users/jeremyhoward/Desktop/C++Code/SamplingGamates/17_GenoDiver_V2/GenoDiverFiles") ## Change <br>
			###################### <br>
			### Phase Persistance  ### <br>
			###################### <br>
			df <- read.table(file="Phase_Persistance_Generation",sep=" ",header=FALSE,skip=1); ## Read in file ## <br>
			## Grab the last generation ## <br>
			df <- matrix(unlist(strsplit(t(df[20,-c(1)]),":")),ncol=5,byrow=TRUE) <br>
			distance <- c("0-0.5","0.5-1.0","1.0-1.5","1.5-2.0","2.0-2.5") <br>
			for(i in 1:5) <br>
			{ <br>
			&emsp; temp <- data.frame(cbind(df[,i],rep(distance[i],nrow(df)),1:nrow(df)),stringsAsFactors = FALSE) <br>
			&emsp; names(temp) <- c("phase","window","Generation") <br>
			&emsp; if(i == 1){plotdf <- temp;rm(temp)} <br>
			&emsp; if(i > 1){plotdf <- rbind(plotdf,temp); rm(temp)} <br>
			} <br>
			plotdf$phase <- as.numeric(paste(plotdf$phase)); plotdf$Generation <- as.numeric(paste(plotdf$Generation)); <br>
			ggplot(plotdf, aes(x=Generation, y = phase, colour = window)) + geom_line(size = 1) +  <br>
  			labs(title = "Phase Persistence", x = "Generation", y = "Phase Persistence with Generation 20") + theme_bw() + <br>
  			theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12),legend.position="bottom",axis.text=element_text(size=10)) <br>
  		</div>
	</div>		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example9cFigures.png" ></p>
	</div>
	
	<div id = "parameterclass">
	 	<font color="black"> Run of Homozygosity (ROH) </font>
	</div>
	<br>
	<div id = "textbox">
		<div id = "text">
			&emsp; Within the "Summary_Statistics_ROH_Freq" file the frequency of a SNP being in a continuous run of homozygosity (ROH) of at least 5 Mb was calculated. This was used due to 
			longer ROH arise in the genome as a result of recent inbreeding. The R code below outlines the changes and heterogeneity of the frequency of a SNP being in a ROH as 
			selection proceeds.
		</div>
	</div>
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2); require(gridExtra); require(grid) <br>
			setwd("/Users/jeremyhoward/Desktop/C++Code/SamplingGamates/17_GenoDiver_V2/GenoDiverFiles") ## Change <br>
			############################## <br>
			### ROH across the genome  ### <br>
			############################## <br>
			df <- read.table(file="Summary_Statistics_ROH_Freq",sep=" ",header=TRUE); <br>
			df$row <- 1:nrow(df) <br>
			generations <- c(5,10,15,20) <br>
			## Save ggplot2 plots in list <br>
			Plots <- vector('list',4); ## Stores all of plots ## <br>
			for(i in 1:length(generations)) <br>
			{ <br> 
  			&emsp; tempdf <- df[ ,c(1,2,7,(i+2))]; names(tempdf) <- c("chr","pos","row","freq") <br>
  			&emsp; tempdf <- tempdf[which(tempdf$freq != -5), ] <br>
  			&emsp; agg.snp <- aggregate(row ~ chr, data=tempdf, FUN=mean); breaks.snp <- round(agg.snp$row, 0) <br>
  			&emsp; title <- paste("Generation ",generations[i],sep="")<br>
  			&emsp; Plots[[i]] <- ggplot(tempdf, aes(x=row, y = freq,colour = as.factor(chr))) + geom_point(size = 0.01) + ylim(0.0,0.40) +<br>
     		&emsp; &emsp; labs(title = title, x = "Chromosome", y = "Frequency of a SNP in a ROH of 5 Mb ") + theme_bw() +<br>
      		&emsp; &emsp; scale_color_manual(values=c("red","blue","red")) + scale_x_continuous(breaks= breaks.snp,labels=c("1","2","3")) +<br>
      		&emsp; &emsp; theme(plot.title = element_text(size = 3.5,hjust = 0.5),axis.text=element_text(size=3),axis.title=element_text(size=3.5),<br>
            &emsp; &emsp; legend.text = element_text(size = 4.0),legend.position="none")<br>
			}<br>
			## Plot each one<br>
			Plots[[1]]<br>
			Plots[[2]]<br>
			Plots[[3]]<br>
			Plots[[4]]<br>
  		</div>
	</div>		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example9dFigures.png" ></p>
	</div>
	
	<div id = "parameterclass">
	 	<font color="black"> Window Variance </font>
	</div>
	<br>
	<div id = "textbox">
		<div id = "text">
			&emsp; Within the "WindowAdditiveVariance" file the true additive genetic variance within 1 Mb non-overlapping across the genome is generated. Each row corresponds to 
			a generation and the locations are outlined in the first row. The R code below outlines the change in the amount of additive genetic variance across the genome as 
			selection proceeds.
		</div>
	</div>
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2); require(gridExtra); require(grid) <br>
			setwd("/Users/jeremyhoward/Desktop/C++Code/SamplingGamates/17_GenoDiver_V2/GenoDiverFiles") ## Change <br>
			########################################## <br>
			### Window Variance across the genome  ### <br> 
			########################################## <br>
			df <- read.table(file="WindowAdditiveVariance",sep=" ",header=FALSE); <br> 
			## Generate df with location <br>
			Locations <- data.frame(cbind(t(df[1,c(2:ncol(df))]),1:(ncol(df)-1)),stringsAsFactors = FALSE) <br>
			Locations$chr <- matrix(unlist(strsplit(Locations$X1,"_")),ncol=2,byrow=TRUE)[,1] <br>
			Locations$pos <- matrix(unlist(strsplit(Locations$X1,"_")),ncol=2,byrow=TRUE)[,2] <br>
			Locations <- Locations[ ,c(2,3,4)]; <br>
			names(Locations) <- c("row","chr","pos") <br>
			Locations$row <- as.numeric(paste(Locations$row)); Locations$chr <- as.numeric(paste(Locations$chr)); <br>
			## which generations to grab <br>
			generations <- c(5,15,20) <br>
			## Make dataframe for ggplot2 <br>
			Plots <- vector('list',length(generations)); ## Stores all of plots ## <br>
			for(i in 1:length(generations)) <br>
			{<br>
 			 &emsp; tempdf <- cbind(Locations,t(df[generations[i]+2,c(2:ncol(df))])); <br>
			 &emsp; tempdf[,4] <- as.numeric(paste(tempdf[,4])) <br>
 			 &emsp; tempdf[,4] <- (tempdf[,4] / sum(tempdf[,4])) * 100; ## Plot as proportion of total <br>
 			 &emsp; names(tempdf)[4] <- "varianceexp" <br>
 			 &emsp; agg.snp <- aggregate(row ~ chr, data=tempdf, FUN=mean); breaks.snp <- round(agg.snp$row, 0) <br>
 			 &emsp; title <- paste("Generation ",generations[i],sep="") <br>
  			 &emsp; Plots[[i]] <- ggplot(tempdf, aes(x=row, y = varianceexp,colour = as.factor(chr))) + geom_point(size = 0.01) + ylim(0.0,20) +<br>
      		 &emsp; &emsp;labs(title = title, x = "Chromosome", y = "Variance Explained (%) ") + theme_bw() +<br>
      		 &emsp; &emsp;scale_color_manual(values=c("red","blue","red")) + scale_x_continuous(breaks= breaks.snp,labels=c("1","2","3")) + <br>
      		 &emsp; &emsp;theme(plot.title = element_text(size = 3.5,hjust = 0.5),axis.text=element_text(size=3),axis.title=element_text(size=3.5),<br>
       	     &emsp; &emsp;legend.text = element_text(size = 4.0),legend.position="none")<br>
  			}<br>
			Plots[[1]]<br>
			Plots[[2]]<br>
			Plots[[3]]<br>
  		</div>
	</div>		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example9eFigures.png" ></p>
	</div>
	
	<div id = "homepage">
		<a href="{{site.baseurl}}/index.html">Home Page</a>
	</div>
	
	
	

</html>

