<!DOCTYPE html>
<html>
	<head> 
    	 <meta charset="utf-8">
     	<link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
     	<title> GenoDiver Livestock Simulation Software </title>
     	<!-- Global site tag (gtag.js) - Google Analytics -->
		<script async src="https://www.googletagmanager.com/gtag/js?id=UA-110857544-1"></script>
		<script>
		  window.dataLayer = window.dataLayer || [];
		  function gtag(){dataLayer.push(arguments);}
		  gtag('js', new Date());

		  gtag('config', 'UA-110857544-1');
		</script>
		<script>
			function ParameterFileSummary()
			{
				var x = document.getElementById("parametersummary");
				if(x.style.display === "none"){
				x.style.display = "block";
				} else {x.style.display = "none";}
			}
		</script>
   	</head> 
	<body>

	<div id = "header" >
   		<div id = "maintitle" >  GenoDiver </div>
   		<div id = "subtitle" > A Coalescent Forward in Time Simulation Toolkit </div>	
   		<div id = "navigation" > 
   			<a href = "index.html"> Home </a> &emsp;
   			<a href = "about.html"> About </a> &emsp;      
   			<a href = "parameters.html"> Parameters </a> &emsp;
   			<a href = "outputfiles.html"> Output Files </a> &emsp;
   			<a href = "examples.html"> Examples </a>
   		</div>
   	</div>
	<br>
	
	<div id = "parameterclass">
	 	<font color="black"> Running the Simulation Software </font>
	</div>	 	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; At the current time, executable files are only available for Linux operating environments and are located on the GitHub page for
	 		<a href="https://github.com/jeremyhoward/Geno-Diver">Geno-Diver</a>. It has also been tested on a MAC operating environment and the manual 
	 		 describes how to compile the program. To run the program place, "GenoDiver", "macs" and "msformatter" executable files in the folder where 
	 		 the program will run. Before running the program, the file permissions need to be checked. After verifying the permissions, a parameter file, 
	 		 outlined below, will need to be generated and placed in the same folder  as the previous three executable files. A parameter file can be 
	 		 generated using any text editor. <br> <br> 
	 		&emsp; The simulation program reads the parameter file by searching for keywords that are capitalized and followed by a colon. Therefore any
			phrase that does not meet the search criteria is ignored when initializing parameters within the program. Also, if you want to comment out a
			parameter just add "!!" within the key word and the program will skip over it. For example to skip over the "SEED" parameter just replace it with 
			"SE!!ED" and it won't recognize the parameter any more. <br> <br>	  
		</div>
	</div>
	
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			−−−−−−−|     Running the Program Example   |−−−−−−− <br>
			−| General |−  <br>
			START: sequence <br>
			SEED: 1500 <br>
			−| Genome & Marker |− <br>
			CHR: 3 <br>
			CHR_LENGTH: 150 150 150 <br>
			NUM_MARK: 4000 4000 4000 <br>
			QTL: 150 150 150 <br>
			−| Population |− <br>
			FOUNDER_Effective_Size: Ne70 <br>
			MALE_FEMALE_FOUNDER: 50 400 random 3 <br>
			VARIANCE_A: 0.10 <br>
			−| Selection |− <br>
			GENERATIONS: 15 <br>
			INDIVIDUALS: 50 0.2 400 0.2 <br>
			PROGENY: 1 <br>
			SELECTION: ebv high <br>
			EBV_METHOD: pblup <br>
			CULLING: ebv 5 <br>
			-| Mating |- <br>
			MATING: random125 simu_anneal <br>
		</div>
	</div>
	<br>
	<div class = "parameterclass-container">
			<a onclick = "ParameterFileSummary()" > <b> Parameter File Summary </b> 
		<div id = "parametersummary">
			<div id = "textparam">
				&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases (Mb). The simulated genome has a high degree 
				of short- range LD (Ne70). The SNP panel contains 12,000 markers (i.e. 4,000 mark- ers per chromosome). For each chromosome, 150 randomly 
				placed QTL and zero FTL mutations were generated. The quantitative trait simulated has a narrow sense heritability of 0.10 and only additive 
				effects are generated (i.e. no dominance). The phenotypic variance is by default set at 1.0, and therefore the residual variance is 0.90. The 
				founder population consisted of 50 males and 400 females. For each generation, a total of 50 males and 400 females are in the population. Random 
				selecton of progeny and culling of parents was conducted for 3 generations in order to build up the pedigree. A total of 10 and 80 (0.2 replacement 
				rate) male and female parents, re- spectively, are culled and replaced by new progeny each generation. After 3 generations, animals with a high EBV 
				are selected or culled each gener- ation. Fifteen generations are simulated. The EBV are estimated using an animal model with a pedigree-based 
				relationship matrix. Each mating pair produced one progeny. Parents with a pedigree-based relationships greater than 0.125 were avoided, and 
				this was optimized based on the simulated annealing method. <br>
		 	</div>
		</div>
	</div>
	<br>
	<div id = "textbox">
		<div id = "text">
			&emsp; To run the program type “./GenoDiver” and the name of your parameter file. For example, if the parameter file is named “parameterfile”,
			 the program is run by typing in “./GenoDiver parameterfile”. During the simulation, the program outputs minimal comments on the progress. A more
			 thorough description of the program's status is printed to the log file (i.e. “log_file.txt”). <br> <br>
			&emsp; After the program has finished it is a strongly recommended to check the parameters initialized at the top of the log file (i.e. "log_file.txt") within
			the output folder. The log file contains a large amount of information and is a great tool to ensure that the parameters and outcome
			 of the simulation is what is intended. If the program is not running correctly the log file should provide knowledge on why and
			  where the simulation crashed or exited.  <br> <br>
			&emsp; The files are by default placed in the "GenoDiverFiles" directory. If the "OUTPUTFOLDER" option is utilized, the files will be placed 
			in the user-specified directory. Below is a screenshot of the files that are generated from the simulation software based on the parameter
			 file outlined above.
		</div>
	</div>
	
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example1a.png" ></p>
	</div>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; A number of files are generated, but only a few are needed to generate summary statistics on the simulation program and include: <br>
			<ul>
  				<li> <b>Master_DataFrame</b>: File with phenotype, inbreeding and pedigree information across all animals.</li>
  				<li> <b>Master_Genotypes.gz</b>: File with genotype information for each animal. </li>
  				<li> <b>QTL_new_old_Class</b>: File with information for each QTL/FTL mutation. </li>
  				<li> <b>Marker_Map</b>: Location of markers.</li>
  				<li> <b>Summary_Statistics_DataFrame_Performance</b>: Statistics by generation for performance metrics. </li>
  				<li> <b>Summary_Statistics_DataFrame_Inbreeding</b>: Statistics by generation for inbreeding metrics. </li>
  				<li> <b>Summary_Statistics_QTL</b>: Statistics by generation for QTL/FTL metrics. </li>
			</ul>	
			
			&emsp; Utilizing the R code outlined below the following plots were generated from the output files. The following example utilizes traditional pedigree information when 
			predicting breeding values for individuals. To test the impact of genomic information on the plots outlined below change the EBV_METHOD to 'gblup' and use the R code below
			to see the difference.
			<br>
		</div>
	</div>
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			<span class="functioncall">rm</span><span class="keyword">(</span><span class="argument">list</span> <span class="argument">=</span> <span class="functioncall">ls</span><span class="keyword">(</span><span class="keyword">)</span><span class="keyword">)</span><span class="keyword">;</span> <span class="functioncall">gc</span><span class="keyword">(</span><span class="keyword">)</span> <br>
			<span class="functioncalla">library</span><span class="keyword">(</span><span class="symbol">ggplot2</span><span class="keyword">); </span> <span class="functioncalla">library</span><span class="keyword">(</span><span class="symbol">tidyverse</span><span class="keyword">)</span> <br>
			<span class="comment">## Change</span> <br>
			<span class="functioncall">setwd</span><span class="keyword">(</span><span class="string">"/Users/jeremyhoward/Desktop/C++Code/18_GenoDiver_V3/GenoDiverFiles/"</span><span class="keyword">)</span> <br>
			<span class="comment">#############################</span> <br>		
			<span class="comment">## Plot True Genetic Value ##</span> <br>
			<span class="comment">#############################</span> <br>
			<span class="symbol">df</span> <span class="assignement">&lt;-</span> <span class="functioncall">read_table2</span><span class="keyword">(</span><span class="argument">file</span><span class="argument">=</span><span class="string">"Summary_Statistics_DataFrame_Performance"</span><span class="keyword">,</span><span class="argument">col_names</span> <span class="argument">=</span> <span class="number">TRUE</span><span class="keyword">,</span><span class="argument">col_type</span> <span class="argument">=</span> <span class="string">"dcccccc"</span><span class="keyword">)</span> <span class="keyword">%&gt;%</span> <br>
  			&emsp; <span class="functioncall">mutate</span><span class="keyword">(</span><span class="symbol">.</span><span class="keyword">,</span><span class="argument">tbv</span> <span class="argument">=</span> <span class="functioncall">as.numeric</span><span class="keyword">(</span><span class="functioncall">matrix</span><span class="keyword">(</span><span class="functioncall">unlist</span><span class="keyword">(</span><span class="functioncall">strsplit</span><span class="keyword">(</span><span class="symbol">tbv</span><span class="keyword">,</span> <span class="string">"[()]"</span><span class="keyword">)</span><span class="keyword">)</span><span class="keyword">,</span> <span class="argument">ncol</span> <span class="argument">=</span> <span class="number">2</span><span class="keyword">,</span> <span class="argument">byrow</span> <span class="argument">=</span> <span class="number">TRUE</span><span class="keyword">)</span><span class="keyword">[</span><span class="keyword">,</span> <span class="number">1</span><span class="keyword">]</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">%&gt;%</span> <br>
  			&emsp; <span class="functioncall">select</span><span class="keyword">(</span><span class="symbol">Generation</span><span class="keyword">,</span><span class="symbol">tbv</span><span class="keyword">)</span> <br>
  			<br>
  			<span class="functioncall">ggplot</span><span class="keyword">(</span><span class="symbol">df</span><span class="keyword">,</span> <span class="functioncall">aes</span><span class="keyword">(</span><span class="argument">x</span> <span class="argument">=</span> <span class="symbol">Generation</span><span class="keyword">,</span> <span class="argument">y</span> <span class="argument">=</span> <span class="symbol">tbv</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">geom_line</span><span class="keyword">(</span><span class="argument">size</span> <span class="argument">=</span> <span class="number">1</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">ggtitle</span><span class="keyword">(</span><span class="string">"Genetic Trend"</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">theme_bw</span><span class="keyword">(</span><span class="keyword">)</span> <span class="keyword">+</span> <br>
 			&emsp; <span class="functioncall">theme</span><span class="keyword">(</span><span class="argument">plot.title</span> <span class="argument">=</span> <span class="functioncall">element_text</span><span class="keyword">(</span><span class="argument">hjust</span> <span class="argument">=</span> <span class="number">0.5</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">ylab</span><span class="keyword">(</span><span class="string">"Mean True Breeding Value "</span><span class="keyword">)</span> <br>
			<span class="comment">##############################</span> <br>	
			<span class="comment">## Plot Pedigree Inbreeding ##</span> <br>	
			<span class="comment">##############################</span> <br>	
			<span class="symbol">df</span> <span class="assignement">&lt;-</span> <span class="functioncall">read_table2</span><span class="keyword">(</span><span class="argument">file</span><span class="argument">=</span><span class="string">"Summary_Statistics_DataFrame_Inbreeding"</span><span class="keyword">,</span><span class="argument">col_names</span> <span class="argument">=</span> <span class="number">TRUE</span><span class="keyword">,</span><span class="argument">col_type</span> <span class="argument">=</span> <span class="string">"dcccccccccccccc"</span><span class="keyword">)</span> <span class="keyword">%&gt;%</span> <br>
  			&emsp; <span class="functioncall">mutate</span><span class="keyword">(</span><span class="symbol">.</span><span class="keyword">,</span><span class="argument">ped_f</span> <span class="argument">=</span> <span class="functioncall">as.numeric</span><span class="keyword">(</span><span class="functioncall">matrix</span><span class="keyword">(</span><span class="functioncall">unlist</span><span class="keyword">(</span><span class="functioncall">strsplit</span><span class="keyword">(</span><span class="symbol">ped_f</span><span class="keyword">,</span> <span class="string">"[()]"</span><span class="keyword">)</span><span class="keyword">)</span><span class="keyword">,</span> <span class="argument">ncol</span> <span class="argument">=</span> <span class="number">2</span><span class="keyword">,</span> <span class="argument">byrow</span> <span class="argument">=</span> <span class="number">TRUE</span><span class="keyword">)</span><span class="keyword">[</span><span class="keyword">,</span> <span class="number">1</span><span class="keyword">]</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">%&gt;%</span> <br>
  			&emsp; <span class="functioncall">select</span><span class="keyword">(</span><span class="symbol">Generation</span><span class="keyword">,</span><span class="symbol">ped_f</span><span class="keyword">)</span> <br>
			<br>
			<span class="functioncall">ggplot</span><span class="keyword">(</span><span class="symbol">df</span><span class="keyword">,</span> <span class="functioncall">aes</span><span class="keyword">(</span><span class="argument">x</span> <span class="argument">=</span> <span class="symbol">Generation</span><span class="keyword">,</span> <span class="argument">y</span> <span class="argument">=</span> <span class="symbol">ped_f</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">geom_line</span><span class="keyword">(</span><span class="argument">size</span> <span class="argument">=</span> <span class="number">1</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">ggtitle</span><span class="keyword">(</span><span class="string">"Inbreeding Trend"</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">theme_bw</span><span class="keyword">(</span><span class="keyword">)</span> <span class="keyword">+</span> <br>
 			&emsp; <span class="functioncall">theme</span><span class="keyword">(</span><span class="argument">plot.title</span> <span class="argument">=</span> <span class="functioncall">element_text</span><span class="keyword">(</span><span class="argument">hjust</span> <span class="argument">=</span> <span class="number">0.5</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">ylab</span><span class="keyword">(</span><span class="string">"Mean Pedigree Inbreeding "</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">xlab</span><span class="keyword">(</span><span class="string">"Generation"</span><span class="keyword">)</span> <br>
			<span class="comment">#############################</span> <br>
			<span class="comment">## Allele Frequency Change ##</span> <br>
			<span class="comment">#############################</span> <br>
			<span class="symbol">df</span> <span class="assignement">&lt;-</span> <span class="functioncall">read_table2</span><span class="keyword">(</span><span class="argument">file</span><span class="argument">=</span><span class="string">"QTL_new_old_Class"</span><span class="keyword">,</span><span class="argument">col_names</span> <span class="argument">=</span> <span class="number">TRUE</span><span class="keyword">,</span><span class="argument">col_type</span> <span class="argument">=</span> <span class="string">"dcccccc"</span><span class="keyword">)</span> <br>
			<span class="comment">## split apart frequencies ##</span> <br>
			<span class="symbol">freq</span> <span class="assignement">&lt;-</span> <span class="functioncall">matrix</span><span class="keyword">(</span><span class="functioncall">unlist</span><span class="keyword">(</span><span class="functioncall">strsplit</span><span class="keyword">(</span><span class="symbol">df</span><span class="keyword">$</span><span class="symbol">Freq</span><span class="keyword">,</span> <span class="string">"_"</span><span class="keyword">)</span><span class="keyword">)</span><span class="keyword">,</span> <span class="argument">ncol</span> <span class="argument">=</span> <span class="number">16</span><span class="keyword">,</span> <span class="argument">byrow</span> <span class="argument">=</span> <span class="number">TRUE</span><span class="keyword">)</span> <br>
			<span class="symbol">freq</span> <span class="assignement">&lt;-</span> <span class="functioncall">apply</span><span class="keyword">(</span><span class="symbol">freq</span><span class="keyword">,</span> <span class="number">2</span><span class="keyword">,</span> <span class="symbol">as.numeric</span><span class="keyword">)</span> <br>
			<span class="comment">## y axis is in terms of change in the favorable direction ##</span> <br>
			<span class="comment">## grab ones with a positive effect and subtract of initial value #</span> <br>
			<span class="symbol">X</span> <span class="assignement">&lt;-</span> <span class="functioncall">which</span><span class="keyword">(</span><span class="symbol">df</span><span class="keyword">$</span><span class="symbol">Additive_Selective</span> <span class="keyword">&gt;</span> <span class="number">0</span><span class="keyword">)</span> <br>
			<span class="symbol">freq</span><span class="keyword">[</span><span class="symbol">X</span><span class="keyword">,</span> <span class="keyword">]</span> <span class="assignement">&lt;-</span> <span class="symbol">freq</span><span class="keyword">[</span><span class="symbol">X</span><span class="keyword">,</span> <span class="keyword">]</span> <span class="keyword">-</span> <span class="symbol">freq</span><span class="keyword">[</span><span class="symbol">X</span><span class="keyword">,</span> <span class="number">1</span><span class="keyword">]</span> <br>
			<span class="comment">## grab ones with a negative effect and subtract of initial value #</span> <br>
			<span class="symbol">X</span> <span class="assignement">&lt;-</span> <span class="functioncall">which</span><span class="keyword">(</span><span class="symbol">df</span><span class="keyword">$</span><span class="symbol">Additive_Selective</span> <span class="keyword">&lt;</span> <span class="number">0</span><span class="keyword">)</span> <br>
			<span class="symbol">freq</span><span class="keyword">[</span><span class="symbol">X</span><span class="keyword">,</span> <span class="keyword">]</span> <span class="assignement">&lt;-</span> <span class="keyword">(</span><span class="symbol">freq</span><span class="keyword">[</span><span class="symbol">X</span><span class="keyword">,</span> <span class="number">1</span><span class="keyword">]</span> <span class="keyword">-</span> <span class="symbol">freq</span><span class="keyword">[</span><span class="symbol">X</span><span class="keyword">,</span> <span class="keyword">]</span><span class="keyword">)</span> <br>
			<span class="comment">## get mean by generation</span> <br>
			<span class="symbol">plotdf</span> <span class="assignement">&lt;-</span> <span class="functioncall">data.frame</span><span class="keyword">(</span><span class="functioncall">cbind</span><span class="keyword">(</span><span class="functioncall">c</span><span class="keyword">(</span><span class="number">0</span><span class="keyword">:</span><span class="number">15</span><span class="keyword">)</span><span class="keyword">,</span> <span class="functioncall">colMeans</span><span class="keyword">(</span><span class="symbol">freq</span><span class="keyword">)</span><span class="keyword">)</span><span class="keyword">)</span> <br>
			<span class="functioncall">names</span><span class="keyword">(</span><span class="symbol">plotdf</span><span class="keyword">)</span> <span class="assignement">&lt;-</span> <span class="functioncall">c</span><span class="keyword">(</span><span class="string">"gen"</span><span class="keyword">,</span> <span class="string">"freq"</span><span class="keyword">)</span> <br>
			<br>
			<span class="functioncall">ggplot</span><span class="keyword">(</span><span class="symbol">plotdf</span><span class="keyword">,</span> <span class="functioncall">aes</span><span class="keyword">(</span><span class="argument">x</span> <span class="argument">=</span> <span class="symbol">gen</span><span class="keyword">,</span> <span class="argument">y</span> <span class="argument">=</span> <span class="symbol">freq</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">geom_line</span><span class="keyword">(</span><span class="argument">size</span> <span class="argument">=</span> <span class="number">1.0</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">ggtitle</span><span class="keyword">(</span><span class="string">"Favorable Allele Frequency Change"</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">theme_bw</span><span class="keyword">(</span><span class="keyword">)</span> <span class="keyword">+</span> <br>
  			&emsp; <span class="functioncall">theme</span><span class="keyword">(</span><span class="argument">plot.title</span> <span class="argument">=</span> <span class="functioncall">element_text</span><span class="keyword">(</span><span class="argument">hjust</span> <span class="argument">=</span> <span class="number">0.5</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">ylab</span><span class="keyword">(</span><span class="string">"Change in allele frequency since Genertion 0 "</span><span class="keyword">)</span> <span class="keyword">+</span> <br>
  			&emsp; <span class="functioncall">xlab</span><span class="keyword">(</span><span class="string">"Generation"</span><span class="keyword">)</span> <br>
		</div>
	</div>
	<br>
		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example1Figures.png" ></p>
	</div>
		
	<div id ="genodiverimage">
	  <p> <img src="{{ site.baseurl }}/images/GenoDiver.png" > </p>
	</div>
	
	</body>
</html>

