<!DOCTYPE html>
<html>

	<head> 
     <meta charset="utf-8">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
   	 </head> 

	<div id = "title">
	  <p>
	  		<p><img src="{{ site.baseurl }}/images/Geno_Diver_Logo_Small.png" ></p>
	  </p>
	</div>
	
	
	<div id = "parameterclass">
	 	<font color="black"> Comparing Different EBV Prediction Methods </font>
	</div>	 	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; A bash script is outlined below that illustrates how to loop through different simulation scenarios. The following script may take
	 		a couple of hours to finish depending on the computing environment. The difference between the scenarios is how estimated breeding values 
	 		(EBV) are estimated across individuals. The three methods to predict EBV are: a pedigree-based best linear unbiased prediction (BLUP) animal 
	 		model, a genomic-based BLUP animal model and a Bayesian spike and slab marker effects model (BayesC). All three scenarios utilize the same 
	 		founder sequence, which is generated by first using the 'sequence' option for the "START" parameter and then the remaining scenarios use 
	 		the 'founder' option for the "START" parameter. In order to generate multiple replicates within each scenario, the "NREP" parameter is 
	 		utilized and 10 replicates were generated within each scenario. The replicates are saved within the "GenoDiverFiles" directory for each 
	 		scenario. If the directory isn't renamed, the next scenario will replace the files. Therefore, after a scenario is done the directory is 
	 		renamed within the bash script.
	 		<br> <br>	  
		</div>
	</div>
	
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			##-------------------------------------------------------------------------------- <br>
			## First generate parameter file. Put parameters to change <br>
			## at the very bottom of the file. That way you can use the <br>
			## 'head' linux command for the parameters that change and <br>
			## use the 'echo' linux command to add any new parameters. <br>
			##-------------------------------------------------------------------------------- <br>
			## In case 4_Exmaple.txt is still a file ## <br>
			rm -rf 4_Example4.txt || TRUE <br>
			<br>
			# Parameters that are the same # <br>
			echo '-| General |-' >> 4_Example4.txt <br>
			echo 'SEED: 1500' >> 4_Example4.txt <br>
			echo 'NREP: 10' >> 4_Example4.txt <br>
			echo '−| Genome & Marker |−' >> 4_Example4.txt <br>
			echo 'CHR: 3' >> 4_Example4.txt <br>
			echo 'CHR_LENGTH: 150 150 150' >> 4_Example4.txt <br>
			echo 'NUM_MARK: 4000 4000 4000' >> 4_Example4.txt <br>
			echo 'QTL: 50 50 50' >> 4_Example4.txt <br>
			echo '−| Population |−' >> 4_Example4.txt <br>
			echo 'FOUNDER_Effective_Size: Ne100_Scen1' >> 4_Example4.txt <br>
			echo 'MALE_FEMALE_FOUNDER: 50 400 random 5' >> 4_Example4.txt <br>
			echo 'VARIANCE_A: 0.35' >> 4_Example4.txt <br>
			echo 'VARIANCE_D: 0.05' >> 4_Example4.txt <br>
			echo '−| Selection |−' >> 4_Example4.txt <br>
			echo 'GENERATIONS: 5' >> 4_Example4.txt <br>
			echo 'INDIVIDUALS: 50 0.2 400 0.2' >> 4_Example4.txt <br>
			echo 'PROGENY: 1' >> 4_Example4.txt <br>
			echo 'SELECTION: ebv high' >> 4_Example4.txt <br>
			echo 'CULLING: ebv 8' >> 4_Example4.txt <br>
			echo 'MATING: random125 simu_anneal' >> 4_Example4.txt <br>
			<br>
			##-------------------------------------------------------------------------------- <br>
			## First do PBLUP based EBV prediction <br>
			##-------------------------------------------------------------------------------- <br>
			echo 'START: sequence' >> 4_Example4.txt <br>
			echo 'EBV_METHOD: pblup' >> 4_Example4.txt <br>
			## Run GenoDiver ## <br>
			./GenoDiver 4_Example4.txt <br>
			## rename replicate output to rep_pblup ## <br>
			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_pblup <br>
			<br>
			##-------------------------------------------------------------------------------- <br>
			## Then do GBLUP based EBV prediction <br>
			##-------------------------------------------------------------------------------- <br>
			## Remove parameters that are changing ## <br>
			head -n 20 4_Example4.txt > 4_Example4a.txt <br>
			mv ./4_Example4a.txt ./4_Example4.txt <br>
			## echo in new paramters ## <br>
			echo 'START: founder' >> 4_Example4.txt <br>
			echo 'EBV_METHOD: gblup' >> 4_Example4.txt <br>
			## Run GenoDiver ## <br>
			./GenoDiver 4_Example4.txt <br>
			## rename replicate output to rep_pblup ## <br>
 			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_gblup <br>
			<br>
			##-------------------------------------------------------------------------------- <br>
			## Then do BAYESC based EBV prediction <br>
			##-------------------------------------------------------------------------------- <br>
			## Remove parameters that are changing ## <br>
			head -n 20 4_Example4.txt > 4_Example4a.txt <br>
			mv ./4_Example4a.txt ./4_Example4.txt <br>
			## echo in new parameters <br>
			echo 'START: founder' >> 4_Example4.txt <br>
			echo 'EBV_METHOD: bayes' >> 4_Example4.txt <br>
			echo 'BAYESOPTIONS: BayesC 10000 2000 6000 1000 5 fix 0.10' >> 4_Example4.txt <br>
			## Run GenoDiver ## <br>
			./GenoDiver 4_Example4.txt <br>
			## rename replicate output to rep_bayesC ## <br>
			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_bayesC <br>
		</div>
	</div>
	
	<br> <br>
	
	<div id = "parametersummary">
		<b> Parameter File Summary </b> <br>
		<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of 
			short-range LD (Ne100_Scen1). The SNP panel contains 12,000 markers (i.e. 4,000 markers per chromosome). For each chromosome, fifty randomly 
			placed QTL and zero FTL mutations were generated. The narrow and broad sense heritability for the trait is 0.35 and 0.40, 
			respectively. The phenotypic variance is by default set at 1.0, and therefore the residual variance is 0.6. The founder population consisted 
			of 50 males and 400 females. For each generation, a total of 50 males and 400 females are in the population. A total of 10 and 80 (0.2 
			replacement rate) male and female parents, respectively, are culled and replaced by new progeny each generation. Random selecton of progeny 
			and culling of parents was conducted for 5 generations. After 5 generations, animals with a high EBV were selected or culled each generation. 
			The EBV are estimated using a pedigree-based BLUP, genomic-based BLUP or Bayesian spike and slab marker effects model (BayesC). Each 
			mating pair produced one progeny. Parents that had pedigree-based relationships greater than 0.125 were avoided, and this was optimized 
			based on the simulated annealing method. <br>
		 </div>
	</div>
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; Outlined below is a more detailed explanation of the major differences in how EBVS are calculated across the
			different scenarios:
			<ul>
  				<li> <font color="Navy">pblup</font>: The EBV are estimated based on a relationship matrix derived from pedigree information. All animals 
  				are utilized when generating the relationship matrix.</li>
  				<li> <font color="Navy">gblup</font>: The EBV are estimated based on a relationship matrix derived from genomic information (G). When 
  				generating the relationship matrix all animals are utilized along with all of the markers. Therefore, the assumption is that all markers 
  				have an impact on the phenotype. Furthermore, allele frequencies utilized to construct G are estimated from the founder population. </li>
  				<li> <font color="Navy">bayes</font>: The EBV are estimated based on bayesian marker effects model, which estimates the effect of each 
  				marker utilizing an MCMC approach. An animals EBV is generated by the summation of an animals genotype times the estimated marker effect
  				 across all markers. When the 'bayes' option is specified, the "BAYESOPTIONS" parameter must be included in the parameter file. The model 
  				 used is referred to as BayesC, which is a spike and slab marker effects model. The 5 values after specifying which type of bayesian model 
  				 to implement refers to how many iterations to run the MCMC chain and the thinning rate. The first time the BayesC model is implemented 
  				 (i.e. Generation 6), a total of 10,000 MCMC iterations are generated with the first 2,000 discarded as burn-in. The following generations 
  				 then use the marker effects from the previous generation as starting values and a total of 6,000 MCMC iterations are generated with the 
  				 first 1,000 discarded as burn-in. Lastly, a thinning rate of 5 is utilized. In BayesC it assumed only a proportion of the markers have 
  				 a non-null effect, while the remaining don't have an effect. In this case we fixed the proportion and we assumed 10 % of the markers 
  				 have an effect. One of the downfalls with MCMC sampling is that it does increase the computation time. Therefore you will notice that 
  				 when using the bayes options the computation time will dramatically increase. </li>
			</ul>	 
			&emsp; The important files are saved within the renamed replicate folder within each scenario. The replicate number is appended to the file name.
			The bash script above renames the replicate directories to "reps_pblup", "reps_gblup" and "reps_bayesC" for the pblup, gblup and bayes EBV estimation 
			scenarios, respectively. A screenshot of the files within the "GenoDiverFiles" is below. The directories are in blue and the files are in white.
		</div>
	</div>
	<br>
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example6.png" ></p>
	</div>
	<br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; Utilizing the R code outlined below the following plot was generated to illustrate how to loop through each scenario and generate plots
			that describe the impact of EBV estimation method on the genetic trend. <br>
		</div>
	</div>
	
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2) <br>
			wd <- "/Users/jeremyhoward/Desktop/GenoDiverFiles/" ## Change <br>
			## the directory name for each scenario ## <br>
			scen <- c("reps_pblup","reps_gblup","reps_bayesC") <br>
			## Number of replicates simulated ## <br>
			reps <- c(1500:1509) <br> <br>
			################################################################## <br>
			## Loop through and grab metric across scenarios and replicates ## <br>
			################################################################## <br>
			for(i in 1:length(scen)) <br>
			{ <br>
			&emsp; for(j in 1:length(reps)) <br>
			&emsp; { <br>
    		&emsp; &emsp; filename <- paste(wd,scen[i],"/Summary_Statistics_DataFrame_Performance_",reps[j],sep="") <br>
    		&emsp; &emsp; df <- read.table(file=filename,header=TRUE,colClasses=c("numeric",rep("character",6))) # read in file <br>
    		&emsp; &emsp; df$Method <- unlist(strsplit(scen[i],"_"))[2]  # save method <br>
    		&emsp; &emsp; df$Rep <- reps[j];  # save replicate <br>
    		&emsp; &emsp; df$Value <- matrix(unlist(strsplit(df[, 4],"\\(")),ncol=2,byrow=TRUE)[,1] # grab mean gv by generation <br>
    		&emsp; &emsp; if(j == 1 & i == 1){summarytable <- df[ ,c(8,9,1,10)]} <br>
    		&emsp; &emsp; if(j > 1 | i > 1){summarytable <- rbind(summarytable,df[ ,c(8,9,1,10)]);} <br>
  			&emsp; } <br>
			} <br>
			## generate mean and sd by generation and method <br>
			summarytable$Value <- as.numeric(paste(summarytable$Value)) <br>
			means <- aggregate(Value ~ Generation + Method, data=summarytable,FUN=mean) <br>
			sds <- aggregate(Value ~ Generation + Method, data=summarytable,FUN=sd) <br> <br>
			################################################# <br>
			## Plot Genetic Trend across Different Methods ## <br>
			################################################# <br>
			plotdf <- cbind(means,sds[,3]); rm(means,sds) <br>
			names(plotdf) <- c("Generation","Method","Mean","SD") <br>
			pd <- position_dodge(0.5) <br>
			ggplot(plotdf, aes(x=Generation, y=Mean, colour=Method)) + <br>
			&emsp; geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD), colour="black", width=.4, size = 0.5, position=pd) + <br>
			&emsp; geom_point(size=2.0) + geom_line(size=0.50) + theme_bw() + <br>
  			&emsp; labs(title = "Genetic Trend \n(+/- 1 SD)", x = "Generation", y = "Mean True Breeding Value") + <br>
  			&emsp; theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12), <br>
  			&emsp; legend.position="bottom",axis.text=element_text(size=10)) <br>
		</div>
	</div>
	<br>
		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example6Figures.png" ></p>
	</div>
				
	<div id = "homepage">
		<a href="{{site.baseurl}}/index.html">Home Page</a>
	</div>
	
	
	

</html>

