<!DOCTYPE html>
<html>

	<head> 
     <meta charset="utf-8">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
   	 </head> 

	<div id = "title">
	  <p>
	  		<p><img src="{{ site.baseurl }}/images/Geno_Diver_Logo_Small.png" ></p>
	  </p>
	</div>
	
	
	<div id = "parameterclass">
	 	<font color="black"> Running the Simulation Software </font>
	</div>	 	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; At the current time, executable files are only available for Linux operating environments and are located on the GitHub page for
	 		<a href="https://github.com/jeremyhoward/Geno-Diver">Geno-Diver</a>. To run the program place, "GenoDiver", "macs" and "msformatter"
	 		 executable files in the folder where the program will run. Before running the program, the file permissions need to be checked.
	 		 After verifying the permissions, a parameter file, outlined below, will need to be generated and placed in the same folder 
	 		 as the previous three executable files. A parameter file can be generated using any text editor or downloaded from the
	 		  <a href="https://github.com/jeremyhoward/Geno-Diver">Geno-Diver</a> GitHub page. <br> <br>
	 		 &emsp; The simulation program reads the parameter file by searching for keywords that are capitalized and followed by a colon. Therefore any
			phrase that does not meet the search criteria is ignored when initializing parameters within the program. Also, if you want to comment out a
			parameter just add "!!" within the key word and the program will skip over it. For example to skip over the "SEED" parameter just replace it with 
			"SE!!ED" and it won't recognize the parameter any more. <br> <br>	  
		</div>
	</div>
	
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			−−−−−−−|     Running the Program Example   |−−−−−−− <br>
			−| General |−  <br>
			START: sequence <br>
			SEED: 1501 <br>
			−| Genome & Marker |− <br>
			CHR: 3 <br>
			CHR_LENGTH: 150 150 150 <br>
			NUM_MARK: 4000 4000 4000 <br>
			QTL: 50 50 50 <br>
			−| Population |− <br>
			FOUNDER_Effective_Size: Ne70 <br>
			MALE_FEMALE_FOUNDER: 50 400 random 0 <br>
			VARIANCE_A: 0.35 <br>
			VARIANCE_D: 0.05 <br>
			−| Selection |− <br>
			GENERATIONS: 15 <br>
			INDIVIDUALS: 50 0.2 400 0.2 <br>
			PROGENY: 1 <br>
			SELECTION: ebv high <br>
			EBV_METHOD: pblup <br>
			CULLING: ebv 5 <br>
			-| Mating |- <br>
			MATING: random125 simu_anneal <br>
		</div>
	</div>
	
	<br> <br>
	
	<div id = "parametersummary">
		<b> Parameter File Summary </b> <br>
		<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of short-range LD 
			(Ne70). The SNP panel contains 12,000 marker (i.e. 4,000 markers per chromosome). For each chromosome, 50 randomly placed QTL were generated, and no FTL 
			mutations were generated. The narrow and broad sense heritability for the trait is 0.35 and 0.40, respectively. The phenotypic variance is by default set 
			at 1.0, and therefore the residual variance is 0.6. The founder population consisted of 50 males and 400 females. For each generation, a total of 50 males 
			and 400 females are in the population. A total of 10 and 80 (0.2 replacement rate) male and female parents, respectively, are culled and replaced by new 
			progeny each generation. A total of 15 generations were simulated. Animals with a high estimated breeding value (EBV) were selected or culled each 
			generation. The EBV are solved using an animal model based on a pedigree-based relationship matrix. Each mating pair produced one progeny. Parents that 
			had pedigree-based relationships greater than 0.125 were not avoided, and this was optimized based on the simulated annealing method. <br>
		 </div>
	</div>
	<br>
	<div id = "textbox">
		<div id = "text">
			&emsp; To run the program type “./GenoDiver” and the name of your parameter file. For example, if the parameter file is named “parameterfile”,
			 the program is run by typing in “./GenoDiver parameterfile”. During the simulation, the program outputs minimal comments on the progress. A more
			 thorough description of the program's status is printed to the log file (i.e. “log_file.txt”). <br> <br>
			&emsp; After the program has finished it is a strongly recommended to check the parameters initialized at the top of the log file (i.e. "log_file.txt") within
			the output folder. The log file contains a large amount of information and is a great tool to ensure that the parameters and outcome
			 of the simulation is what is intended. If the program is not running correctly the log file should provide knowledge on why and
			  where the simulation crashed or exited.  <br> <br>
			&emsp; The files are by default placed in the "GenoDiverFiles" directory. If the "OUTPUTFOLDER" option is utilized, the files will be placed 
			in the user-specified directory. Below is a screenshot of the files that are generated from the simulation software based on the parameter
			 file outlined above.
		</div>
	</div>
	
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example1a.png" ></p>
	</div>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; A number of files are generated, but only a few are needed to generate summary statistics on the simulation program and include: <br>
			<ul>
  				<li>"Master_DataFrame": File with phenotype, inbreeding and pedigree information across all animals.</li>
  				<li>"Master_Genotypes": File with genotype information for each animal. </li>
  				<li>"QTL_new_old_Class": File with information for each QTL/FTL mutation. </li>
  				<li>"Marker_Map": Location of markers.</li>
  				<li>"Summary_Statistics_DataFrame_Performance": Summary statistics by generation on performance metrics. </li>
  				<li>"Summary_Statistics_DataFrame_Inbreeding": Summary statistics by generation on inbreeding metrics. </li>
  				<li>"Summary_Statistics_QTL": Summary statistics by generation QTL/FTL metrics. </li>
			</ul>	
			
			&emsp; Utilizing the R code outlined below the following plots were generated from the output files. <br>
		</div>
	</div>
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2) <br>
			setwd("/Users/jeremyhoward/Desktop/C++Code/SamplingGamates/16_GenoDiver_V2/GenoDiverFiles") ## Change <br> <br>
			############################# <br>
			## Plot True Genetic Value ## <br>
			############################# <br>
			df <- read.table(file="Summary_Statistics_DataFrame_Performance",sep=" ",header=TRUE,colClasses = c("numeric",rep("character",6))) <br>
			plotdf <- data.frame(cbind(as.numeric(df[,1]),as.numeric(matrix(unlist(strsplit(df$gv,'[()]')),ncol=2,byrow=TRUE)[,1]))) <br>
			names(plotdf) <- c("gen","tgv") <br>
			ggplot(plotdf, aes(x=gen,y=tgv)) + geom_line() + <br>
  			&emsp; ggtitle("Genetic Trend") +  theme_bw() + theme(plot.title = element_text(hjust = 0.5)) + <br>
  			&emsp; ylab ("Mean True Genetic Value ") + xlab ("Generation") <br> <br>
			############################## <br>
			## Plot Pedigree Inbreeding ## <br>
			############################## <br>
			df <- read.table(file="Summary_Statistics_DataFrame_Inbreeding",sep=" ",header=TRUE,colClasses = c("numeric",rep("character",14))) <br>
			plotdf <- data.frame(cbind(as.numeric(df[,1]),as.numeric(matrix(unlist(strsplit(df$ped_f,'[()]')),ncol=2,byrow=TRUE)[,1]))) <br>
			names(plotdf) <- c("gen","pedinb") <br>
			ggplot(plotdf, aes(x=gen,y=pedinb)) + geom_line() +  <br>
  			&emsp; ggtitle("Inbreeding Trend") +  theme_bw() + theme(plot.title = element_text(hjust = 0.5)) + <br>
  			&emsp; ylab ("Mean Pedigree Inbreeding ") + xlab ("Generation")  <br> <br>
 			############################# <br>
			## Allele Frequency Change ## <br>
			############################# <br>
			df <- read.table(file="QTL_new_old_Class",sep=" ",header=TRUE,colClasses = c("numeric",rep("character",6))) <br>
			freq <- matrix(unlist(strsplit(df$Freq,'_')),ncol=16,byrow=TRUE) <br>
			freq <- apply(freq,2,as.numeric) <br>
			## y axis is in terms of change in the favorable direction ## <br>
			## grab ones with a positive effect and subtract of initial value # <br>
			X <- which(df$Additive_Selective > 0) <br>
			freq[X, ] <- freq[X,1] - freq[X,] <br>
			## grab ones with a negative effect and subtract of initial value # <br>
			X <- which(df$Additive_Selective < 0) <br>
			freq[X, ] <- (freq[X,1] - freq[X,]);  <br>
			## get mean by generation <br>
			plotdf <- data.frame(cbind(c(0:15),colMeans(freq))) <br>
			names(plotdf) <- c("gen","freq") <br> 
			ggplot(plotdf, aes(x=gen,y=freq)) + geom_line() +  <br>
			&emsp; ggtitle("Favorable Allele Frequency Change") +  theme_bw() + theme(plot.title = element_text(hjust = 0.5)) + <br>
			&emsp; ylab ("Change in allele frequency since Genertion 0 ") + xlab ("Generation")  <br>
		</div>
	</div>
	<br>
		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example1Figures.png" ></p>
	</div>
		
	<div id = "homepage">
		<a href="{{site.baseurl}}/index.html">Home Page</a>
	</div>
	
	
	

</html>

