<!DOCTYPE html>
<html>
	<head> 
    	 <meta charset="utf-8">
     	<link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
     	<title> GenoDiver Livestock Simulation Software </title>
     	<!-- Global site tag (gtag.js) - Google Analytics -->
		<script async src="https://www.googletagmanager.com/gtag/js?id=UA-110857544-1"></script>
		<script>
		  window.dataLayer = window.dataLayer || [];
		  function gtag(){dataLayer.push(arguments);}
		  gtag('js', new Date());

		  gtag('config', 'UA-110857544-1');
		</script>
		<script>
			function ParameterFileSummary()
			{
				var x = document.getElementById("parametersummary");
				if(x.style.display === "none"){
				x.style.display = "block";
				} else {x.style.display = "none";}
			}
		</script>
   	</head> 
	<body>

	<div id = "header" >
   		<div id = "maintitle" >  GenoDiver </div>
   		<div id = "subtitle" > A Coalescent Forward in Time Simulation Toolkit </div>	
   		<div id = "navigation" > 
   			<a href = "index.html"> Home </a> &emsp;
   			<a href = "about.html"> About </a> &emsp;      
   			<a href = "parameters.html"> Parameters </a> &emsp;
   			<a href = "outputfiles.html"> Output Files </a> &emsp;
   			<a href = "examples.html"> Examples </a>
   		</div>
   	</div>
	<br>
	
	<div id = "parameterclass">
	 	<font color="black"> Differential Sire Contribution by Age </font>
	</div>	 	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; The parameter file outlined below illustrates how to simulate a population where older animals are assigned a larger number of mating 
	 		pairs, compared to younger animals. This type of scenario is generated by utilizing the "PARITY_MATES_DIST" parameter with the following values 
	 		"2.0 1.0". These two values are utilized to generate the distribution of mating pairs. A Beta distribution, which is parameterized by two parameters, 
	 		is used to generate the distribution of mating pairs. A beta distribution was utilized in order to allow for a wide range of mating scenarios. For 
	 		example, to generate a scenario where younger animals are assigned a larger number of mating pairs compared to older animals, the parameters need to 
	 		be changed to "1.0 2.0". The number of mating pairs by age class are generated by splitting the cumulative distribution function (CDF) into quadrants 
	 		based on the number of age classes that occur within a generation. The total number of mating pairs within an age class is the proportion that falls 
	 		within the CDF quadrant for a given age class.
	 		<br> <br>	  
		</div>
	</div>
	
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			−−−−−−−|    Differential Sire Contribution by Age   |−−−−−−− <br>
			−| General |− <br>
			START: sequence <br>
			SEED: 1501 <br>
			−| Genome & Marker |− <br>
			CHR: 3 <br>
			CHR_LENGTH: 150 150 150 <br>
			NUM_MARK: 4000 4000 4000 <br>
			QTL: 150 150 150 <br>
			−| Population |− <br>
			FOUNDER_Effective_Size: Ne70 <br>
			MALE_FEMALE_FOUNDER: 50 400 random 3 <br>
			VARIANCE_A: 0.35 <br>
			−| Selection |− <br>
			GENERATIONS: 15 <br>
			INDIVIDUALS: 50 0.2 400 0.2 <br>
			PROGENY: 1 <br>
			PARITY_MATES_DIST: 2.0 1.0 <br>
			SELECTION: ebv high <br>
			EBV_METHOD: pblup <br>
			CULLING: ebv 5 <br>
			-| Mating |- <br> 
			MATING: random125 simu_anneal <br>
		</div>
	</div>
	<br>
	<div class = "parameterclass-container">
		<a onclick = "ParameterFileSummary()" > <b> Parameter File Summary </b> 
		<div id = "parametersummary">
			<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of short-range LD 
			(Ne70). The SNP panel contains 12,000 markers (i.e. 4,000 markers per chromosome). For each chromosome, 150 randomly placed QTL and zero FTL mutations 
			were generated. The quantitative trait simulated has a narrow sense heritability of 0.35 and only additive effects are generated (i.e. no dominance). 
			The phenotypic variance is by default set at 1.0, and therefore the residual variance is 0.65. The founder population consisted of 50 males and 400 females. 
			Random selecton of progeny and culling of parents was conducted for 3 generations. For each generation, a total of 50 males and 400 females are in the 
			population. A total of 10 and 80 (0.2 replacement rate) male and female parents, respectively, are culled and re- placed by new progeny each generation. 
			After 3 generations, animals with a high EBV were selected or culled each generation. A total of 15 gen- erations were simulated. The EBV are estimated 
			using an animal model with a pedigree-based relationship matrix. Each mating pair produced one progeny. The mating distribution was skewed so that older 
			animals had more mating pairs than younger animals. Parents that had pedigree-based relationships greater than 0.125 were avoided, and this was optimized 
			based on the simulated annealing method. <br>
		 	</div>
		</div>
	</div>
	<br>	
	<div id = "textbox">
		<div id = "text">
			&emsp; Once the program has finished, inspection of the log file will provide details on the impact of the "PARITY_MATES_DIST" option. Within the log file 
			(lines 148-169), the mating distribution CDF is illustrated and is outlined below. When generating the number of matings for a given age class for each 
			generation the CDF outlined below is split into quadrants based on the number of age classes that occur within a generation. Lastly, the number of matings 
			within each age class is also outlined in the logfile for each generation.
		</div>
	</div>
	
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example3a.png" ></p>
	</div>
	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; A potential reason for putting this into a simulation is to generate some sires with a large number of progeny, while other sires have 
			very few progeny. The sires that generate a large number of progeny would then have a large impact on the genome of future generations which may include the 
			spread of a lethal/sublethal mutation that the sire(s) carry. <br> <br>
			
			&emsp; Utilizing the R code outlined below the following plot was generated to show the non-linear relationship of number of progeny left by a sire across
			 different ages that a sire left the herd. <br>
		</div>
	</div>
	
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			<span class="functioncall">rm</span><span class="keyword">(</span><span class="argument">list</span><span class="argument">=</span><span class="functioncall">ls</span><span class="keyword">(</span><span class="keyword">)</span><span class="keyword">)</span><span class="keyword">;</span> <span class="functioncall">gc</span><span class="keyword">(</span><span class="keyword">)</span> <br>
			<span class="functioncalla">library</span><span class="keyword">(</span><span class="symbol">ggplot2</span><span class="keyword">)</span><span class="keyword">;</span> <span class="functioncalla">library</span><span class="keyword">(</span><span class="symbol">tidyverse</span><span class="keyword">)</span> <br>
			<span class="comment">## Change </span> <br>
			<span class="functioncall">setwd</span><span class="keyword">(</span><span class="string">"/Users/jeremyhoward/Desktop/C++Code/18_GenoDiver_V3/GenoDiverFiles/"</span><span class="keyword">)</span> <br>
			<span class="comment">##############################################</span> <br>
			<span class="comment">## Plot progeny count by age sire left herd ##</span> <br>
			<span class="comment">##############################################</span> <br>
			<span class="symbol">df</span> <span class="assignement">&lt;-</span> <span class="functioncall">read_table2</span><span class="keyword">(</span><span class="argument">file</span><span class="argument">=</span><span class="string">"Master_DataFrame"</span><span class="keyword">,</span><span class="argument">col_names</span> <span class="argument">=</span> <span class="number">TRUE</span><span class="keyword">,</span><span class="argument">col_type</span> <span class="argument">=</span> <span class="string">"iiiiiiiidddddiiiiddddddddddd"</span><span class="keyword">)</span> <span class="keyword">%&gt;%</span><br>
  			&emsp; <span class="functioncall">filter</span><span class="keyword">(</span><span class="symbol">.</span><span class="keyword">,</span><span class="symbol">Progeny</span> <span class="keyword">&gt;</span> <span class="number">0</span><span class="keyword">)</span> <span class="keyword">%&gt;%</span> <span class="comment">## grab ones who have progeny count &gt; 0</span><br>
  			&emsp; <span class="functioncall">filter</span><span class="keyword">(</span><span class="symbol">.</span><span class="keyword">,</span><span class="symbol">Sex</span> == <span class="number">0</span><span class="keyword">)</span> <span class="keyword">%&gt;%</span> <span class="comment">## grab males only</span><br>
  			&emsp; <span class="functioncall">filter</span><span class="keyword">(</span><span class="symbol">.</span><span class="keyword">,</span><span class="symbol">Gen</span> <span class="keyword">&gt;</span> <span class="number">3</span><span class="keyword">)</span> <span class="comment">## remove animals prior to selection</span><br>
			<br>
			<span class="functioncall">ggplot</span><span class="keyword">(</span><span class="symbol">df</span><span class="keyword">,</span> <span class="functioncall">aes</span><span class="keyword">(</span><span class="argument">x</span><span class="argument">=</span><span class="symbol">Age</span><span class="keyword">,</span> <span class="argument">y</span> <span class="argument">=</span> <span class="symbol">Progeny</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">geom_point</span><span class="keyword">(</span><span class="argument">colour</span> <span class="argument">=</span> <span class="string">"red"</span><span class="keyword">,</span> <span class="argument">size</span> <span class="argument">=</span> <span class="number">3</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">xlab</span><span class="keyword">(</span><span class="string">"Age Sire Left Population"</span><span class="keyword">)</span> <span class="keyword">+ <br>
			&emsp; </span> <span class="functioncall">ylab</span><span class="keyword">(</span><span class="string">"Number of Progeny"</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">ggtitle</span><span class="keyword">(</span><span class="string">"Differential Sire Contributions by Age"</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">theme_bw</span><span class="keyword">(</span><span class="keyword">)</span> <br>
		</div>
	</div>
	<br>
		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example9Figures.png" ></p>
	</div>
				
	<div id ="genodiverimage">
	  <p> <img src="{{ site.baseurl }}/images/GenoDiver.png" > </p>
	</div>
	
	</body>
</html>

