<!DOCTYPE html>
<html>

	<head> 
     <meta charset="utf-8">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
   	 </head> 

	<div id = "title">
	  <p>
	  		<p><img src="{{ site.baseurl }}/images/Geno_Diver_Logo_Small.png" ></p>
	  </p>
	</div>
	
	
	<div id = "parameterclass">
	 	<font color="black"> Differential Sire Contribution by Age </font>
	</div>	 	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; The parameter file outlined below illustrates how to simulate a population where older animals are assigned a larger number of mating 
	 		pairs compared to younger animals. This type of scenario is generated by utilizing the "PARITY_MATES_DIST" parameter with the following values 
	 		"2.0 1.0". These two values are utilized to generate the distribution of mating pairs. A Beta distribution, which is parameterized by two parameters, 
	 		is used to generate the distribution of mating pairs. A beta distribution was utilized in order to allow for a wide range of mating scenarios. For 
	 		example, to generate a scenario where younger animals are assigned a larger number of mating pairs compared to older animals, the parameters need to 
	 		be changed to "1.0 2.0". The number of mating pairs by age class are generated by splitting the cumulative distribution function (CDF) into quadrants 
	 		based on the number of age classes that occur within a generation. The total number of mating pairs within an age class is the proportion that falls 
	 		within the CDF quadrant for a given age class.
	 		<br> <br>	  
		</div>
	</div>
	
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			−−−−−−−|    Differential Sire Contribution by Age   |−−−−−−− <br>
			−| General |− <br>
			START: sequence <br>
			SEED: 1501 <br>
			−| Genome & Marker |− <br>
			CHR: 3 <br>
			CHR_LENGTH: 150 150 150 <br>
			NUM_MARK: 4000 4000 4000 <br>
			QTL: 50 50 50 <br>
			−| Population |− <br>
			FOUNDER_Effective_Size: Ne70 <br>
			MALE_FEMALE_FOUNDER: 50 400 random 0 <br>
			VARIANCE_A: 0.35 <br>
			VARIANCE_D: 0.05 <br>
			−| Selection |− <br>
			GENERATIONS: 15 <br>
			INDIVIDUALS: 50 0.2 400 0.2 <br>
			PROGENY: 1 <br>
			PARITY_MATES_DIST: 2.0 1.0 <br>
			SELECTION: ebv high <br>
			EBV_METHOD: pblup <br>
			CULLING: ebv 5 <br>
			-| Mating |- <br> 
			MATING: random125 simu_anneal <br>
		</div>
	</div>
	
	<br> <br>
	
	<div id = "parametersummary">
		<b> Parameter File Summary </b> <br>
		<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of short-range LD 
			(Ne70). The SNP panel contains 12,000 marker (i.e. 4,000 markers per chromosome). For each chromosome, 50 randomly placed QTL were generated, and no FTL 
			mutations were generated. The narrow and broad sense heritability for the trait is 0.35 and 0.40, respectively. The phenotypic variance is by default set 
			at 1.0, and therefore the residual variance is 0.6. The founder population consisted of 50 males and 400 females. For each generation, a total of 50 males 
			and 400 females are in the population. A total of 10 and 80 (0.2 replacement rate) male and female parents, respectively, are culled and replaced by new 
			progeny each generation. A total of 15 generations were simulated. Animals with a high estimated breeding value (EBV) were selected or culled each 
			generation. The EBV are solved using an animal model based on a pedigree-based relationship matrix. Each mating pair produced one progeny. The mating 
			distribution was skewed so that older animals had more mating pairs than younger animals. Parents that had pedigree-based relationships greater than 0.125 
			were not avoided, and this was optimized based on the simulated annealing method. <br>
		 </div>
	</div>
	<br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; Once the program has finished, inspection of the log file will provide details on the impact of the "PARITY_MATES_DIST" option. Within the log file 
			(lines 148-169), the mating distribution CDF is illustrated and is outlined below. When generating the number of matings for a given age class for each 
			generation the CDF outlined below is split into quadrants based on the number of age classes that occur within a generation. 
		</div>
	</div>
	
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example3a.png" ></p>
	</div>
	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; A potential reasons for putting this into a simulation is to generate some sires with a large number of progeny while other sires have 
			very few progeny. The sires that generate a large number of progeny would then have a large impact on the genome of future generations which may include the 
			spread of a lethal/sublethal mutation that the sire(s) carry. <br> <br>
			
			&emsp; Utilizing the R code outlined below the following plot was generated to show the non-linear relationship of number of progeny left by a sire across
			 different ages at which a sire left the herd. <br>
		</div>
	</div>
	
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2) <br>
			setwd("/Users/jeremyhoward/Desktop/C++Code/SamplingGamates/16_GenoDiver_V2/GenoDiverFiles") ## Change <br>
			############################################## <br>
			## Plot progeny count by age sire left herd ## <br>
			############################################## <br>
			df <- read.table(file="Master_DataFrame",sep=" ",header=TRUE) <br>
			df <- df[which(df$Progeny > 0), ] ## grab ones who have progeny count > 0 <br>
			df <- df[which(df$Sex == 0), ] ## grab males only <br>
			df <- df[which(df$Gen > 1), ] ## remove founders <br>
			ggplot(df, aes(x=Age, y = Progeny)) + geom_point(colour = "red", size = 3) + <br>
			&emsp; xlab("Age Sire Left Population") + ylab("Number of Progeny") +  <br>
  			&emsp; ggtitle("Differential Sire Contributions by Age") + theme_bw()  <br>
		</div>
	</div>
	<br>
		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example3Figures.png" ></p>
	</div>
				
	<div id = "homepage">
		<a href="{{site.baseurl}}/index.html">Home Page</a>
	</div>
	
	
	

</html>

