<!DOCTYPE html>
<html>
	<head> 
    	 <meta charset="utf-8">
     	<link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
     	<title> GenoDiver Livestock Simulation Software </title>
     	<!-- Global site tag (gtag.js) - Google Analytics -->
		<script async src="https://www.googletagmanager.com/gtag/js?id=UA-110857544-1"></script>
		<script>
		  window.dataLayer = window.dataLayer || [];
		  function gtag(){dataLayer.push(arguments);}
		  gtag('js', new Date());

		  gtag('config', 'UA-110857544-1');
		</script>
		<script>
			function ParameterFileSummary()
			{
				var x = document.getElementById("parametersummary");
				if(x.style.display === "none"){
				x.style.display = "block";
				} else {x.style.display = "none";}
			}
		</script>
   	</head> 
	<body>

	<div id = "header" >
   		<div id = "maintitle" >  GenoDiver </div>
   		<div id = "subtitle" > A Coalescent Forward in Time Simulation Toolkit </div>	
   		<div id = "navigation" > 
   			<a href = "index.html"> Home </a> &emsp;
   			<a href = "about.html"> About </a> &emsp;      
   			<a href = "parameters.html"> Parameters </a> &emsp;
   			<a href = "outputfiles.html"> Output Files </a> &emsp;
   			<a href = "examples.html"> Examples </a>
   		</div>
   	</div>
	<br>
		
	<div id = "parameterclass">
	 	<font color="black"> Multiple Progeny Quantitative Trait </font>
	</div>	 	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; The parameter file outlined below illustrates how to simulate a prolific species (i.e. Swine) with a 
	 		threshold on the minimum number of progeny to select within each family. <br> <br>	  
		</div>

	</div>

	<div id = "raisedbox">
		<div id = "raisedboxtext">
			−−−−−−−|     Multiple Progeny Quantitative Trait   |−−−−−−− <br>
			−| General |− <br>
			START: sequence <br>
			SEED: 1501 <br>
			−| Genome & Marker |− <br>
			CHR: 3 <br>
			CHR_LENGTH: 150 150 150 <br>
			NUM_MARK: 4000 4000 4000 <br>
			QTL: 150 150 150 <br>
			−| Population |− <br>
			FOUNDER_Effective_Size: Ne70 <br>
			MALE_FEMALE_FOUNDER: 25 200 random 3 <br>
			VARIANCE_A: 0.35 <br>
			−| Selection |− <br>
			GENERATIONS: 10 <br>
			INDIVIDUALS: 25 0.2 200 0.2 <br>
			PROGENY: 8 <br>
			MAXFULLSIB: 2 <br>
			SELECTION: ebv high <br>
			EBV_METHOD: pblup <br>
			CULLING: ebv 5 <br>
			-| Mating |- <br> 
			MATING: random125 simu_anneal <br>
		</div>
	</div>
	<br>
	<div class = "parameterclass-container">
			<a onclick = "ParameterFileSummary()" > <b> Parameter File Summary </b> 
		<div id = "parametersummary">
			<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases (Mb). The genome simulated has a high degree of short-range LD 
			(Ne70). The SNP panel contains 12,000 markers (i.e. 4,000 markers per chromosome). For each chromosome, 150 randomly placed QTL and zero FTL mutations were 
			generated. The quantitative trait simulated has a narrow sense heritability of 0.35 and only additive effects are generated (i.e. no dominance). The phenotypic 
			variance is by default set at 1.0, and therefore the residual variance is 0.65. The founder population consisted of 25 males and 200 females. Random selecton of 
			progeny and culling of parents was conducted for 3 generations. For each generation, a total of 25 males and 200 females are in the population. A total of 5 and 
			40 (0.2 replacement rate) male and female parents, respectively, are culled and replaced by new progeny each generation. After 3 generations, animals with a high 
			EBV were selected or culled each generation. A total of 10 generations were simulated. The EBV are estimated using an animal model with a pedigree-based 
			relationship matrix. Each mating pair produced eight progeny. Within each full-sib family a maximum of 2 progeny can be selected. Parents that had pedigree-based 
			relationships greater than 0.125 were avoided, and this was optimized based on the simulated annealing method. <br>
		 	</div>
		</div>
	</div>
	<br>
	<div id = "textbox">
		<div id = "text">
			&emsp; Once the program has finished, inspection of the log file will provide details on the impact of the "MAXFULLSIB" option. Within the log file, 
			after the "Begin ebv Selection of offspring" section, the number of times "n" number of siblings were selected within a family is outlined for each 
			generation. This provides an overview of the degree of co-selection of siblings that occur based on the simulation design outlined above. If this option 
			would not have been included, the maximum number of siblings that can be selected within a family would be the total number of offspring generated for each 
			mating pair (i.e. 8).
		</div>
	</div>
	
	<div id ="genodiverimage">
	  <p> <img src="{{ site.baseurl }}/images/GenoDiver.png" > </p>
	</div>
	
	</body>
</html>

