<!DOCTYPE html>
<html>
	<head> 
    	 <meta charset="utf-8">
     	<link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
     	<title> GenoDiver Livestock Simulation Software </title>
     	<!-- Global site tag (gtag.js) - Google Analytics -->
		<script async src="https://www.googletagmanager.com/gtag/js?id=UA-110857544-1"></script>
		<script>
		  window.dataLayer = window.dataLayer || [];
		  function gtag(){dataLayer.push(arguments);}
		  gtag('js', new Date());

		  gtag('config', 'UA-110857544-1');
		</script>
   	</head> 
	<body>

	<div id = "header" >
   		<div id = "maintitle" >  GenoDiver </div>
   		<div id = "subtitle" > A Coalescent Forward in Time Simulation Toolkit </div>	
   		<div id = "navigation" > 
   			<a href = "index.html"> Home </a> &emsp;
   			<a href = "about.html"> About </a> &emsp;      
   			<a href = "parameters.html"> Parameters </a> &emsp;
   			<a href = "outputfiles.html"> Output Files </a> &emsp;
   			<a href = "examples.html"> Examples </a>
   		</div>
   	</div>
	<br>
	
	<div id = "parameterclass">
	 	<font color="black"> Simulating a Fitness and Quantitative Trait </font>
	</div>	 	
	<br> <br>
	
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; The parameter file outlined below illustrates how to simulate a quantitative trait that has a proportion (i.e. 50%) of the quantitative trait loci (QTL) also having 
	 		a fitness effect. The relationship between the QTL with a fitness effect has a positive correlation of 0.20, but the two traits are antagonistic based on the way the fitness 
	 		value of a loci is parameterized. High selection coefficients (s) result in the unfavorable homozygote genotype to be less fit and is described in detail in the 'QTL/FTL 
	 		Distributions' link. As outlined in the previous example, it is important to add a few extra animals in the founder population since a portion will die. If the number of male 
	 		or female founder animals is smaller than the population size it will exit out of the program. Therefore, careful consideration of the number and magnitude of fitness effects 
	 		along with the number of progeny produced per mating pair needs to be carefully considered when constructing the simulation design. <br> <br>
	 	</div>
	</div>

	</div>
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			−−−−−−−|     Quantitative and Fitness Trait   |−−−−−−− <br>
			−| General |− <br>
			START: sequence <br>
			SEED: 1501 <br>
			−| Genome & Marker |− <br>
			CHR: 3 <br>
			CHR_LENGTH: 150 150 150 <br>
			NUM_MARK: 4000 4000 4000 <br>
			QTL: 50 50 50 <br>
			FIT_LETHAL: 15 15 15 <br>
			FIT_SUBLETHAL: 100 100 100 <br>
			−| Population |− <br>
			FOUNDER_Effective_Size: Ne70 <br>
			MALE_FEMALE_FOUNDER: 100 350 random 0 <br>
			VARIANCE_A: 0.20 <br>
			VARIANCE_D: 0.05 <br>
			COVAR: 0.5 0.2 <br>
			−| Selection |− <br>
			GENERATIONS: 20 <br>
			INDIVIDUALS: 50 0.2 250 0.2 <br>
			PROGENY: 1 <br>
			SELECTION: ebv high <br>
			EBV_METHOD: pblup <br>
			CULLING: ebv 5 <br>
			-| Mating |- <br> 
			MATING: random <br>
		</div>
	</div>
	
	<br> <br>
	
	<div id = "parametersummary">
		<b> Parameter File Summary </b> <br>
		<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of 
			short-range LD (Ne70). The SNP panel contains 12,000 markers (i.e. 4,000 markers per chromosome). For each chromosome, 150 randomly 
			placed QTL were generated. Also, 15 lethal and 100 sub-lethal mutations were generated for each chromosome. The narrow and broad sense heritability for the 
			quantitative trait is 0.20 and 0.25, respectively. The phenotypic variance is by default set at 1.0, and therefore the residual variance 
			is 0.75. Half of the QTL also have a fitness effect and the correlation between the additive QTL effects and sub-lethal selection coefficients is 0.20.
			The founder population consisted of 100 males and 350 females. For each generation, a total of 50 males and 250 females are in the population. A total 
			of 10 and 50 (0.2 replacement rate) male and female parents, respectively, are culled and replaced by new progeny each generation. Starting at the first 
			generation animals with a high EBV were selected or culled each generation. he EBV are estimated using an animal model with a pedigree-based relationship matrix.  
			Each mating pair produced one progeny. Parents were mated at random. <br>
		 </div>
	</div>
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; Inspection of the log file will provide summary statistics on the mean selection coefficients and degree of dominance for the lethal and sub-lethal 
			fitness effect along with their associated frequency. The default settings for the lethal mutations result in a high selection coefficient (0.70) and very little 
			dominance (0.001). Therefore the heterozygote is normal, while the unfavorable homozygote has a low fitness value. The settings for the sub-lethal mutations result in a 
			lower selection coefficient (0.02) and moderate degree of dominance (0.30). Therefore the heterozygote now has a reduced fitness value compared to the fittest homozygote. 
			Lastly, the mean allele frequency of the unfavorable allele is higher for sub-lethal compared to lethals. Below is a screenshot of the lines in the log file that display 
			the summary statistics.
		</div>
	</div>

	
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example5a.png" ></p>
	</div>
	
	<div id = "textbox">
		<div id = "text">		
			&emsp; Utilizing the R code outlined below the following plots were generated from the output files. <br>
		</div>
	</div>
	<br>
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			rm(list=ls()); gc() <br>
			library(ggplot2); require(gridExtra); require(grid) <br>
			setwd("/Users/jeremyhoward/Desktop/C++Code/SamplingGamates/17_GenoDiver_V2/GenoDiverFiles") ## Change <br>
			## Read in Inbreeding Summary Statistics File <br>
			df <- read.table(file="Summary_Statistics_DataFrame_Inbreeding",sep=" ",header=TRUE,colClasses=c("numeric",rep("character",14))) <br>
			## Plot Mean Fitness Across Generations ## <br> <br>
			tempdf <- data.frame(cbind(df[,1],matrix(unlist(strsplit(df$fitness,'[()]')),ncol=2,byrow=TRUE)))[, c(1,2)] <br>
			names(tempdf) <- c("Generation","fitness") <br>
			tempdf$Generation <- as.numeric(paste(tempdf$Generation)); <br>
			tempdf$fitness <- as.numeric(paste(tempdf$fitness)); <br>
			Plot1 <- ggplot(tempdf, aes(x=Generation, y = fitness)) + geom_point(colour = "red", size = 3) + <br>
			&emsp; labs(title = "", x = "Generation", y = "Fitness") +  theme_bw() + <br>
  			&emsp; theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12),axis.text=element_text(size=10)) <br> <br>
  			## Plot Average Homozygous Sublethal Per IndividualAcross Generations ## <br>
			tempdf <- data.frame(cbind(df[,1],matrix(unlist(strsplit(df$homozysublethal,'[()]')),ncol=2,byrow=TRUE)))[, c(1,2)] <br>
			names(tempdf) <- c("Generation","homosublethal") <br>
			tempdf$Generation <- as.numeric(paste(tempdf$Generation)); <br>
			tempdf$homosublethal <- as.numeric(paste(tempdf$homosublethal)); <br>
			Plot2 <- ggplot(tempdf, aes(x=Generation, y = homosublethal)) + geom_point(colour = "red", size = 3) + <br>
  			&emsp; labs(title = "", x = "Generation", y = "Average Homozygous Sublethal Per Individual") +  theme_bw() + <br>
  			&emsp; theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12),axis.text=element_text(size=10)) <br> <br>
			## Plot Average Heterozygous Sublethal Per IndividualAcross Generations ## <br>
			tempdf <- data.frame(cbind(df[,1],matrix(unlist(strsplit(df$hetezsublethal,'[()]')),ncol=2,byrow=TRUE)))[, c(1,2)] <br>
			names(tempdf) <- c("Generation","hetezsublethal") <br>
			tempdf$Generation <- as.numeric(paste(tempdf$Generation)); <br>
			tempdf$hetezsublethal <- as.numeric(paste(tempdf$hetezsublethal)); <br>
			Plot3 <- ggplot(tempdf, aes(x=Generation, y = hetezsublethal)) + geom_point(colour = "red", size = 3) + <br>
			&emsp; labs(title = "", x = "Generation", y = "Average Heterozygous Sublethal Per Individual") +  theme_bw() +  <br>
 			&emsp; theme(plot.title = element_text(size = 16,hjust = 0.5),axis.title = element_text(size = 12),axis.text=element_text(size=10)) <br> <br>
			## Save three plots to a png file <br>
			png(file = "Example3.png", width =12, height = 6, units="in", res=300) <br>
			PlotAU <- grid.arrange(arrangeGrob(Plot1,Plot2,Plot3,nrow=1)) <br>
			dev.off() <br>
		</div>
	</div>
	<br>
		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example5Figures.png" ></p>
	</div>
		
	<div id ="genodiverimage">
	  <p> <img src="{{ site.baseurl }}/images/GenoDiver.png" > </p>
	</div>
	
	</body>
</html>

