<!DOCTYPE html>
<html>
	<head> 
    	 <meta charset="utf-8">
     	<link rel="stylesheet" href="{{ site.baseurl }}/css/examples.css">
     	<title> GenoDiver Livestock Simulation Software </title>
     	<!-- Global site tag (gtag.js) - Google Analytics -->
		<script async src="https://www.googletagmanager.com/gtag/js?id=UA-110857544-1"></script>
		<script>
		  window.dataLayer = window.dataLayer || [];
		  function gtag(){dataLayer.push(arguments);}
		  gtag('js', new Date());
		  gtag('config', 'UA-110857544-1');
		</script>
		<script>
			function ParameterFileSummary()
			{
				var x = document.getElementById("parametersummary");
				if(x.style.display === "none"){
				x.style.display = "block";
				} else {x.style.display = "none";}
			}
		</script>
   	</head> 
	<body>

	<div id = "header" >
   		<div id = "maintitle" >  GenoDiver </div>
   		<div id = "subtitle" > A Coalescent Forward in Time Simulation Toolkit </div>	
   		<div id = "navigation" > 
   			<a href = "index.html"> Home </a> &emsp;
   			<a href = "about.html"> About </a> &emsp;      
   			<a href = "parameters.html"> Parameters </a> &emsp;
   			<a href = "outputfiles.html"> Output Files </a> &emsp;
   			<a href = "examples.html"> Examples </a>
   		</div>
   	</div>
	<br>
	
	<div id = "parameterclass">
	 	<font color="black"> Comparing Different EBV Prediction Methods </font>
	</div>	 	
	<br> <br>
	
	<div id = "textbox">
		<div id = "text">
	 		&emsp; A bash script is outlined below that illustrates how to loop through different simulation scenarios. The following script may take
	 		a couple of hours to finish depending on the computing environment. The difference between the scenarios is how estimated breeding values 
	 		(EBV) are estimated across individuals. The four methods to predict EBV are: a pedigree-based best linear unbiased prediction (BLUP) animal 
	 		model, a genomic-based BLUP animal model, a run-of-homozygosity-based BLUP animal model and a Bayesian spike and slab marker effects model 
	 		(BayesC). All three scenarios utilize the same founder sequence, which is generated by first using the 'sequence' option for the "START" 
	 		parameter and then the remaining scenarios use the 'founder' option for the "START" parameter. In order to generate multiple replicates 
	 		within each scenario, the "NREP" parameter is utilized and 15 replicates were generated within each scenario. The replicates are saved 
	 		within the "GenoDiverFiles" directory for each scenario. If the directory isn't renamed, the next scenario will replace the files. Therefore, 
	 		after a scenario is done the directory is renamed within the bash script.
	 		<br> <br>	  
		</div>
	</div>
	
	<div id = "raisedbox">
		<div id = "raisedboxtext">
			<font color="#00FFFF">##-------------------------------------------------------------------- </font> <br>
			<font color="#00FFFF">## First generate parameter file. Put parameters to change at the very </font> <br>
			<font color="#00FFFF">## bottom of the file. That way you can use the 'head' linux command </font> <br>
			<font color="#00FFFF">## for the parameters that change and use the 'echo' linux command to </font> <br>
			<font color="#00FFFF">## add any new parameters. </font> <br>
			<font color="#00FFFF">##-------------------------------------------------------------------- </font> <br>
			<font color="#00FFFF">## In case 4_Example.txt is still a file ## </font> <br>
			rm <font color="#FFB6C1">-rf </font> Example12.txt || TRUE <br>
			<br>
			<font color="#00FFFF"># Parameters that are the same #</font> <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">-| General |-</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">SEED: 1500</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">NREP: 15</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">−| Genome & Marker |−</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">CHR: 3</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">CHR_LENGTH: 150 150 150</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">NUM_MARK: 4000 4000 4000</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">QTL: 200 200 200</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">−| Population |−</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">FOUNDER_Effective_Size: Ne100_Scen1</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">MALE_FEMALE_FOUNDER: 50 400 random 5</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">VARIANCE_A: 0.35</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">−| Selection |−</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">GENERATIONS: 10</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">INDIVIDUALS: 50 0.2 400 0.2</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">PROGENY: 1</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">SELECTION: ebv high</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">CULLING: ebv 8</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">MATING: random125 simu_anneal</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<br>
			<font color="#00FFFF">##--------------------------------------------------------------------</font> <br>
			<font color="#00FFFF">## First do PBLUP based EBV prediction </font><br>
			<font color="#00FFFF">##--------------------------------------------------------------------</font> <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">START: sequence</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">EBV_METHOD: pblup</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#00FFFF">## Run GenoDiver ##</font> <br>
			./GenoDiver Example12.txt <br>
			<font color="#00FFFF">## rename replicate output to rep_pblup ##</font> <br>
			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_pblup <br>
			<br>
			<font color="#00FFFF">##--------------------------------------------------------------------</font> <br>
			<font color="#00FFFF">## Then do GBLUP based EBV prediction</font> <br>
			<font color="#00FFFF">##--------------------------------------------------------------------</font> <br>
			<font color="#00FFFF">## Remove parameters that are changing ##</font> <br>
			head <font color="#FFB6C1">-n 19</font> Example12.txt > Example12a.txt <br>
			mv ./Example12a.txt ./Example12.txt <br>
			<font color="#00FFFF">## echo in new paramters ##</font> <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">START: founder</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">EBV_METHOD: gblup</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#00FFFF">## Run GenoDiver ##</font> <br>
			./GenoDiver Example12.txt <br>
			<font color="#00FFFF">## rename replicate output to rep_gblup ##</font> <br>
 			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_gblup <br>
			<br>
			<font color="#00FFFF">##--------------------------------------------------------------------</font> <br>
			<font color="#00FFFF">## Then do ROHBLUP based EBV prediction</font> <br>
			<font color="#00FFFF">##--------------------------------------------------------------------</font> <br>
			<font color="#00FFFF">## Remove parameters that are changing ##</font> <br>
			head <font color="#FFB6C1">-n 19</font> Example12.txt > Example12a.txt <br>
			mv ./Example12a.txt ./Example12.txt <br>
			<font color="#00FFFF">## echo in new paramters ##</font> <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">START: founder</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">EBV_METHOD: rohblup</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#00FFFF">## Run GenoDiver ##</font> <br>
			./GenoDiver Example12.txt <br>
			<font color="#00FFFF">## rename replicate output to rep_rohblup ##</font> <br>
 			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_rohblup <br>
			<br>			
			<font color="#00FFFF">##--------------------------------------------------------------------</font> <br>
			<font color="#00FFFF">## Then do BAYESC based EBV prediction</font> <br>
			<font color="#00FFFF">##--------------------------------------------------------------------</font> <br>
			<font color="#00FFFF">## Remove parameters that are changing ##</font> <br>
			head <font color="#FFB6C1">-n 19</font> Example12.txt > Example12a.txt <br>
			mv ./Example12a.txt ./Example12.txt <br>
			<font color="#00FFFF">## echo in new parameters</font> <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">START: founder</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">EBV_METHOD: bayes</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#FFFF00">echo '</font><font color="#6A5ACD">BAYESOPTIONS: BayesC 10000 2000 6000 1000 5 fix 0.10</font><font color="#FFFF00">' >> </font> Example12.txt <br>
			<font color="#00FFFF">## Run GenoDiver ##</font> <br>
			./GenoDiver Example12.txt <br>
			<font color="#00FFFF">## rename replicate output to rep_bayesC ##</font> <br>
			mv ./GenoDiverFiles/replicates ./GenoDiverFiles/reps_bayesC <br>
		</div>
	</div>
	
	<br>
	<div class = "parameterclass-container">
		<a onclick = "ParameterFileSummary()" > <b> Parameter File Summary </b> 
		<div id = "parametersummary">
			<div id = "textparam">
			&emsp; Sequence information is generated for three chromosomes with a length of 150 Megabases. The genome simulated has a high degree of 
			short-range LD (Ne100_Scen1). The SNP panel contains 12,000 markers (i.e. 4,000 markers per chromosome). For each chromosome, 200 randomly 
			placed QTL and zero FTL mutations were generated. The quantitative trait simulated has a narrow sense heritability of 0.35 and only additive 
			effects are generated (i.e. no dominance). The phenotypic variance is by default set at 1.0, and therefore the residual variance is 0.65.
			The founder population consisted of 50 males and 400 females. For each generation, a total of 50 males and 400 females are in the population. 
			A total of 10 and 80 (0.2 replacement rate) male and female parents, respectively, are culled and replaced by new progeny each generation. 
			Random selecton of progeny and culling of parents was conducted for 5 generations. After 5 generations, animals with a high EBV were selected 
			or culled each generation. The EBV are estimated using a pedigree-based BLUP, genomic-based BLUP, ROH-based BLUP or Bayesian spike and slab 
			marker effects model (BayesC). Each mating pair produced one progeny. Parents that had pedigree-based relationships greater than 0.125 were 
			avoided, and this was optimized based on the simulated annealing method. <br>
			</div>
		</div>
	</div>
	<br>	
		
	<div id = "textbox">
		<div id = "text">
			&emsp; Outlined below is a more detailed explanation of the major differences in how EBVS are calculated across the
			different scenarios:
			<ul>
  				<li> <font color="Navy">pblup</font>: The EBV are estimated based on a relationship matrix derived from pedigree information. All animals 
  				are utilized when generating the relationship matrix.</li>
  				<li> <font color="Navy">gblup</font>: The EBV are estimated based on a relationship matrix derived from genomic information (G). When 
  				generating the relationship matrix all animals are utilized along with all of the markers. Therefore, the assumption is that all markers 
  				have an impact on the phenotype. Furthermore, allele frequencies utilized to construct G are estimated from the founder population. </li>
  				<li> <font color="Navy">rohblup</font>: The EBV are estimated based on a relationship matrix derived from runs-of-homozygosity (ROH). When 
  				generating the relationship matrix all animals are utilized along with all of the markers. Therefore, the assumption is that all markers 
  				have an impact on the phenotype. </li>
  				<li> <font color="Navy">bayes</font>: The EBV are estimated based on bayesian marker effects model, which estimates the effect of each 
  				marker utilizing an MCMC approach. An animals EBV is generated by the summation of an animals genotype times the estimated marker effect
  				across all markers. When the 'bayes' option is specified, the "BAYESOPTIONS" parameter must be included in the parameter file. The model 
  				used is referred to as BayesC, which is a spike and slab marker effects model. The 5 values after specifying which type of bayesian model 
  				to implement refers to how many iterations to run the MCMC chain and the thinning rate. The first time the BayesC model is implemented 
  				(i.e. Generation 6), a total of 10,000 MCMC iterations are generated with the first 2,000 discarded as burn-in. The following generations 
  				then use the marker effects from the previous generation as starting values and a total of 6,000 MCMC iterations are generated with the 
  				first 1,000 discarded as burn-in. Lastly, a thinning rate of 5 is utilized. In BayesC it assumed only a proportion of the markers have 
  				a non-null effect, while the remaining don't have an effect. In this case we fixed the proportion and we assumed 10 % of the markers 
  				have an effect. One of the downfalls with MCMC sampling is that it does increase the computation time. Therefore you will notice that 
  				when using the bayes options the computation time will dramatically increase. </li>
			</ul>	 
			&emsp; The important files are saved within the renamed replicate folder within each scenario. The replicate number is appended to the file name.
			The bash script above renames the replicate directories to "reps_pblup", "reps_gblup", "reps_rohblup" and "reps_bayesC" for the pblup, gblup, rohblup
			and bayes EBV estimation scenarios, respectively. A screenshot of the files within the "GenoDiverFiles" is below. The directories are in blue and 
			the files are in white.
		</div>
	</div>
	<br>
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example12.png" ></p>
	</div>
	<br>
	
	<div id = "textbox">
		<div id = "text">
			&emsp; Utilizing the R code outlined below the following plot was generated to illustrate how to loop through each scenario and generate plots
			that describe the impact of EBV estimation method on the genetic trend. <br>
		</div>
	</div>
	
	<br>
	<div id = "Rcode">
		<b> R-Code </b> <br>
		<div id = "textRcode">
			<span class="functioncall">rm</span><span class="keyword">(</span><span class="argument">list</span><span class="argument">=</span><span class="functioncall">ls</span><span class="keyword">(</span><span class="keyword">)</span><span class="keyword">)</span><span class="keyword">;</span> <span class="functioncall">gc</span><span class="keyword">(</span><span class="keyword">)</span> <br>
			<span class="functioncalla">library</span><span class="keyword">(</span><span class="symbol">ggplot2</span><span class="keyword">)</span><span class="keyword">;</span> <span class="functioncalla">library</span><span class="keyword">(</span><span class="symbol">tidyverse</span><span class="keyword">)</span> <br>
			<span class="comment">## Change </span> <br>
			<span class="symbol">wd</span> <span class="assignement">&lt;-</span> <span class="string">"/Users/jeremyhoward/Documents/39_GenoDiver_C++Code/WebsiteExamples/Example12/"</span> <br>
			<span class="symbol">scen</span> <span class="assignement">&lt;-</span> <span class="functioncall">c</span><span class="keyword">(</span><span class="string">"reps_pblup"</span><span class="keyword">,</span><span class="string">"reps_gblup"</span><span class="keyword">,</span><span class="string">"reps_rohblup"</span><span class="keyword">,</span><span class="string">"reps_bayesC"</span><span class="keyword">)</span> <span class="comment">## the directory name for each scenario ##</span> <br>
			<span class="symbol">reps</span> <span class="assignement">&lt;-</span> <span class="functioncall">c</span><span class="keyword">(</span><span class="number">1500</span><span class="keyword">:</span><span class="number">1514</span><span class="keyword">)</span>  <span class="comment">## Number of replicates simulated ## </span> <br>
			<span class="comment">################################################################## </span> <br>
			<span class="comment">## Loop through and grab metric across scenarios and replicates ## </span> <br>
			<span class="comment">################################################################## </span> <br>
			<span class="keyworda">for</span><span class="keyword">(</span><span class="symbol">i</span> <span class="keyword">in</span> <span class="number">1</span><span class="keyword">:</span><span class="functioncall">length</span><span class="keyword">(</span><span class="symbol">scen</span><span class="keyword">)</span><span class="keyword">)</span> <br>
			<span class="keyword">{</span> <br>
  			&emsp; <span class="keyworda">for</span><span class="keyword">(</span><span class="symbol">j</span> <span class="keyword">in</span> <span class="number">1</span><span class="keyword">:</span><span class="functioncall">length</span><span class="keyword">(</span><span class="symbol">reps</span><span class="keyword">)</span><span class="keyword">)</span> <br>
  			&emsp; <span class="keyword">{</span> <br>
    		&emsp;&emsp;&emsp; <span class="symbol">filename</span> <span class="assignement">&lt;-</span> <span class="functioncall">paste</span><span class="keyword">(</span><span class="symbol">wd</span><span class="keyword">,</span><span class="symbol">scen</span><span class="keyword">[</span><span class="symbol">i</span><span class="keyword">]</span><span class="keyword">,</span><span class="string">"/Summary_Statistics_DataFrame_Performance_"</span><span class="keyword">,</span><span class="symbol">reps</span><span class="keyword">[</span><span class="symbol">j</span><span class="keyword">]</span><span class="keyword">,</span><span class="argument">sep</span><span class="argument">=</span><span class="string">""</span><span class="keyword">)</span><br>
    		&emsp;&emsp;&emsp; <span class="symbol">df</span> <span class="assignement">&lt;-</span> <span class="functioncall">read_table2</span><span class="keyword">(</span><span class="argument">file</span><span class="argument">=</span><span class="symbol">filename</span><span class="keyword">,</span><span class="argument">col_names</span> <span class="argument">=</span> <span class="number">TRUE</span><span class="keyword">,</span><span class="argument">col_type</span> <span class="argument">=</span> <span class="string">"dcccccc"</span><span class="keyword">)</span> <span class="keyword">%&gt;%</span><br>
      		&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp; <span class="functioncall">mutate</span><span class="keyword">(</span><span class="argument">tgv</span> <span class="argument">=</span> <span class="functioncall">as.numeric</span><span class="keyword">(</span><span class="functioncall">matrix</span><span class="keyword">(</span><span class="functioncall">unlist</span><span class="keyword">(</span><span class="functioncall">strsplit</span><span class="keyword">(</span><span class="symbol">tgv</span><span class="keyword">,</span> <span class="string">"[()]"</span><span class="keyword">)</span><span class="keyword">)</span><span class="keyword">,</span> <span class="argument">ncol</span> <span class="argument">=</span> <span class="number">2</span><span class="keyword">,</span> <span class="argument">byrow</span> <span class="argument">=</span> <span class="number">TRUE</span><span class="keyword">)</span><span class="keyword">[</span><span class="keyword">,</span> <span class="number">1</span><span class="keyword">]</span><span class="keyword">)</span><span class="keyword">,</span><br>
            &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp; <span class="argument">Method</span> <span class="argument">=</span> <span class="functioncall">unlist</span><span class="keyword">(</span><span class="functioncall">strsplit</span><span class="keyword">(</span><span class="symbol">scen</span><span class="keyword">[</span><span class="symbol">i</span><span class="keyword">]</span><span class="keyword">,</span><span class="string">"_"</span><span class="keyword">)</span><span class="keyword">)</span><span class="keyword">[</span><span class="number">2</span><span class="keyword">]</span><span class="keyword">,</span> <span class="argument">Rep</span> <span class="argument">=</span> <span class="symbol">reps</span><span class="keyword">[</span><span class="symbol">j</span><span class="keyword">]</span><span class="keyword">)</span> <span class="keyword">%&gt;%</span> <br>
            &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp; <span class="functioncall">select</span><span class="keyword">(</span><span class="symbol">Generation</span><span class="keyword">,</span><span class="symbol">Method</span><span class="keyword">,</span><span class="symbol">Rep</span><span class="keyword">,</span><span class="symbol">tgv</span><span class="keyword">)</span> <br>
    		&emsp;&emsp;&emsp; <span class="keyworda">if</span><span class="keyword">(</span><span class="symbol">j</span> == <span class="number">1</span> <span class="keyword">&amp;</span> <span class="symbol">i</span> == <span class="number">1</span><span class="keyword">)</span><span class="keyword">{</span><span class="symbol">summarytable</span> <span class="assignement">&lt;-</span> <span class="symbol">df</span><span class="keyword">}</span> <br>
    		&emsp;&emsp;&emsp; <span class="keyworda">if</span><span class="keyword">(</span><span class="symbol">j</span> <span class="keyword">&gt;</span> <span class="number">1</span> <span class="keyword">|</span> <span class="symbol">i</span> <span class="keyword">&gt;</span> <span class="number">1</span><span class="keyword">)</span><span class="keyword">{</span><span class="symbol">summarytable</span> <span class="assignement">&lt;-</span> <span class="functioncall">rbind</span><span class="keyword">(</span><span class="symbol">summarytable</span><span class="keyword">,</span><span class="symbol">df</span><span class="keyword">)</span><span class="keyword">;</span><span class="keyword">}</span> <br>
  			&emsp; <span class="keyword">}</span> <br>
			<span class="keyword">}</span> <br>
			<span class="comment">## generate mean and sd by generation and method </span> <br>
			<span class="symbol">means</span> <span class="assignement">&lt;-</span> <span class="functioncall">aggregate</span><span class="keyword">(</span><span class="symbol">tgv</span> <span class="keyword">~</span> <span class="symbol">Generation</span> <span class="keyword">+</span> <span class="symbol">Method</span><span class="keyword">,</span> <span class="argument">data</span><span class="argument">=</span><span class="symbol">summarytable</span><span class="keyword">,</span><span class="argument">FUN</span><span class="argument">=</span><span class="symbol">mean</span><span class="keyword">)</span> <br>
			<span class="symbol">sds</span> <span class="assignement">&lt;-</span> <span class="functioncall">aggregate</span><span class="keyword">(</span><span class="symbol">tgv</span> <span class="keyword">~</span> <span class="symbol">Generation</span> <span class="keyword">+</span> <span class="symbol">Method</span><span class="keyword">,</span> <span class="argument">data</span><span class="argument">=</span><span class="symbol">summarytable</span><span class="keyword">,</span><span class="argument">FUN</span><span class="argument">=</span><span class="symbol">sd</span><span class="keyword">)</span> <br>
			<span class="comment">################################################# </span> <br>
			<span class="comment">## Plot Genetic Trend across Different Methods ## </span> <br>
			<span class="comment">################################################# </span> <br>
			<span class="symbol">plotdf</span> <span class="assignement">&lt;-</span> <span class="functioncall">cbind</span><span class="keyword">(</span><span class="symbol">means</span><span class="keyword">,</span><span class="symbol">sds</span><span class="keyword">[</span><span class="keyword">,</span><span class="number">3</span><span class="keyword">]</span><span class="keyword">)</span><span class="keyword">;</span> <span class="functioncall">rm</span><span class="keyword">(</span><span class="symbol">means</span><span class="keyword">,</span><span class="symbol">sds</span><span class="keyword">)</span> <br>
			<span class="functioncall">names</span><span class="keyword">(</span><span class="symbol">plotdf</span><span class="keyword">)</span> <span class="assignement">&lt;-</span> <span class="functioncall">c</span><span class="keyword">(</span><span class="string">"Generation"</span><span class="keyword">,</span><span class="string">"Method"</span><span class="keyword">,</span><span class="string">"Mean"</span><span class="keyword">,</span><span class="string">"SD"</span><span class="keyword">)</span> <br>
			<span class="symbol">pd</span> <span class="assignement">&lt;-</span> <span class="functioncall">position_dodge</span><span class="keyword">(</span><span class="number">0.20</span><span class="keyword">)</span> <br>
			<br>
			<span class="functioncall">ggplot</span><span class="keyword">(</span><span class="symbol">plotdf</span><span class="keyword">,</span> <span class="functioncall">aes</span><span class="keyword">(</span><span class="argument">x</span><span class="argument">=</span><span class="functioncall">as.factor</span><span class="keyword">(</span><span class="symbol">Generation</span><span class="keyword">)</span><span class="keyword">,</span> <span class="argument">y</span><span class="argument">=</span><span class="symbol">Mean</span><span class="keyword">,</span> <span class="argument">group</span><span class="argument">=</span><span class="symbol">Method</span><span class="keyword">,</span> <span class="argument">colour</span><span class="argument">=</span><span class="symbol">Method</span><span class="keyword">)</span><span class="keyword">)</span> <span class="keyword">+</span> <br>
  			&emsp; <span class="functioncall">geom_errorbar</span><span class="keyword">(</span><span class="functioncall">aes</span><span class="keyword">(</span><span class="argument">ymin</span><span class="argument">=</span><span class="symbol">Mean</span><span class="keyword">-</span><span class="symbol">SD</span><span class="keyword">,</span> <span class="argument">ymax</span><span class="argument">=</span><span class="symbol">Mean</span><span class="keyword">+</span><span class="symbol">SD</span><span class="keyword">)</span><span class="keyword">,</span> <span class="argument">colour</span><span class="argument">=</span><span class="string">"black"</span><span class="keyword">,</span> <span class="argument">width</span><span class="argument">=</span><span class="number">.4</span><span class="keyword">,</span> <span class="argument">size</span> <span class="argument">=</span> <span class="number">0.5</span><span class="keyword">,</span> <span class="argument">position</span><span class="argument">=</span><span class="symbol">pd</span><span class="keyword">)</span> <span class="keyword">+</span> <br>
  			&emsp; <span class="functioncall">geom_point</span><span class="keyword">(</span><span class="argument">size</span><span class="argument">=</span><span class="number">2.0</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">geom_line</span><span class="keyword">(</span><span class="argument">size</span><span class="argument">=</span><span class="number">0.50</span><span class="keyword">)</span> <span class="keyword">+</span> <span class="functioncall">theme_bw</span><span class="keyword">(</span><span class="keyword">)</span> <span class="keyword">+</span> <br>
  			&emsp; <span class="functioncall">labs</span><span class="keyword">(</span><span class="argument">title</span> <span class="argument">=</span> <span class="string">"Genetic Trend \n(+/- 1 SD)"</span><span class="keyword">,</span> <span class="argument">x</span> <span class="argument">=</span> <span class="string">"Generation"</span><span class="keyword">,</span> <span class="argument">y</span> <span class="argument">=</span> <span class="string">"Mean True Breeding Value"</span><span class="keyword">)</span> <span class="keyword">+</span> <br>
  			&emsp; <span class="functioncall">theme</span><span class="keyword">(</span><span class="argument">plot.title</span> <span class="argument">=</span> <span class="functioncall">element_text</span><span class="keyword">(</span><span class="argument">size</span> <span class="argument">=</span> <span class="number">16</span><span class="keyword">,</span><span class="argument">hjust</span> <span class="argument">=</span> <span class="number">0.5</span><span class="keyword">)</span><span class="keyword">,</span><span class="argument">axis.title</span> <span class="argument">=</span> <span class="functioncall">element_text</span><span class="keyword">(</span><span class="argument">size</span> <span class="argument">=</span> <span class="number">12</span><span class="keyword">)</span><span class="keyword">,</span> <br>
        	&emsp; <span class="argument">legend.position</span><span class="argument">=</span><span class="string">"bottom"</span><span class="keyword">,</span><span class="argument">axis.text</span><span class="argument">=</span><span class="functioncall">element_text</span><span class="keyword">(</span><span class="argument">size</span><span class="argument">=</span><span class="number">10</span><span class="keyword">)</span><span class="keyword">)</span> <br>
		</div>
	</div>
	<br>
		
	<div id = "image">
		<p><img src="{{ site.baseurl }}/images/Example12Figures.png" ></p>
	</div>
				
	<div id ="genodiverimage">
	  <p> <img src="{{ site.baseurl }}/images/GenoDiver.png" > </p>
	</div>
	
	</body>
</html>

