<!DOCTYPE html>
<html>

	<head> 
     <meta charset="utf-8">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/references.css">
   	 </head> 

	<div id = "title">
	  <p>
	  		<p><img src="{{ site.baseurl }}/images/Geno_Diver_Logo_Small.png" ></p>
	  </p>
	</div>
	
	<div id = "parameterclass">
	 	<font color="black"> Summaries of the Files Generated </font>
	</div>	 	
	<br> <br>
	
	
	<div class = "subparameter">
		  <font color="blue">Data Files </font>
	</div>
	<br> 
	<div class="parameter">  	
		<div class="file">	
			log_file.txt: <br>
		</div>
	  	- This file displays a great deal of information on specifics within each generation and it is advisable that one should look over it after you try a new simulation protocol.  <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Master_DataFrame: <br>
		</div>
	  	- This file has a number of metrics for each individual that survived to breeding age and is the primary file to locate useful information.  <br>
	  	<div class="columnheadings">
	  	&emsp; ID: Identification of individual. <br>
	  	&emsp; Sire: Sire ID of individual. <br>
	  	&emsp; Dam: Dam ID of individual. <br>
	  	&emsp; Sex: Sex of individual (0 = male and 1 = female). <br>
	  	&emsp; Gen: Generation the animal was born. <br>
	  	&emsp; Age: Age the animal was removed from the population (i.e. at culling or selection stage). <br>
	  	&emsp; Progeny: Number of progeny. <br>
		&emsp; Dead: Number of dead progeny. <br>
		&emsp; Ped_F: Pedigree based inbreeding metric. <br>
		&emsp; Gen_F: Diagonal of genomic based relationship constructed based on Van Raden (2008). <br>
		&emsp; Hap1_F: Diagonal of haplotype 1 based relationship matrix. <br>
		&emsp; Hap2_F: Diagonal of haplotype 2 based relationship matrix. <br>
		&emsp; Hap3_F: Diagonal of ROH based relationship matrix. <br>
		&emsp; Homolethal: Number of homozygous lethal genotypes. <br>
		&emsp; Heterlethal: Number of heterozygous lethal genotypes. <br>
		&emsp; Homosublethal: Number of homozygous sub-lethal genotypes. <br>
		&emsp; Hetersublethal: Number of heterozygous sub-lethal genotypes. <br>
		&emsp; Letequiv: Lethal equivalent value. <br>
		&emsp; Homozy: Proportion of the genome homozygous. <br>
		&emsp; PropROH: Proportion of the genome in ROH of a given length. <br>
		&emsp; Fitness: Multiplicative Fitness value of the individual. <br>
		&emsp; Phen: Phenotype. <br>
		&emsp; EBV: Estimated Breeding Value. <br>
		&emsp; Acc: Accuracy of EBV. <br>
		&emsp; GV: True genotypic value of individual (Σ (a + d)). <br>
		&emsp; BV: True breeding value of individual (Σ a). <br>
		&emsp; DD: True dominance deviation of individual (Σ d). <br>
		&emsp; R: Residual value of individual. <br>
		</div>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Master_Genotypes: <br>
		</div>
	  	- A file that contains genotypic information for individuals that survived. <br>
	  	<div class="columnheadings">
		&emsp; ID: Identification of individual. <br>
		&emsp; Marker: Marker genotypes of individual (0-11; 2-22; 3-12; 4-21). <br>
		&emsp; QTL: QTL genotypes of individual (0-11; 2-22; 3-12; 4-21). <br>
		</div>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Marker_Map: <br>
		</div>
	  	- The marker map for SNP that were utilized to create a marker panel. <br>
	  	<div class="columnheadings">
		&emsp; chr: Chromosome location. <br>
		&emsp; pos: Nucleotide position of marker. <br>
		</div>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			QTL_new_old_Class: <br>
		</div>
	  	- A file that contains information on QTL effects and frequency across generations. <br>
	  	<div class="columnheadings">
		&emsp; Chr: Chromosome location. <br>
		&emsp; Pos: Nucleotide position of QTL. <br>
		&emsp; Additive Selective: If it is a QTL it refers to the additive effect and if it is a FTL it refers to the selection coefficient. <br>
		&emsp; Dominance: The dominance effect for the QTL or degree of dominance for FTL. <br>
		&emsp; Type: Refers to the type of loci (2 = quantitative trait; 4 = fitness lethal; 5 = fitness sub-lethal). <br>
		&emsp; Gen: Generation at which the mutation occured. <br>
		&emsp; Freq: Gene frequency across generations with a "_" as the delimeter. <br>
		</div>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Low_Fitness: <br>
		</div>
	  	- This file has a number of metrics for each individual that did not survive to breeding age and is the primary file to locate useful information when simulating fitness traits. <br>
	  	<div class="columnheadings">
	  	&emsp; Sire: Sire ID of individual. <br>
	  	&emsp; Dam: Dam ID of individual. <br> 
	  	&emsp; Gen: Generation the animal was born. <br> 
	  	&emsp; Ped_F: Pedigree based inbreeding metric. <br>
	  	&emsp; Gen_F: Diagonal of genomic based relationship constructed based on Van Raden (2008). <br> 
	  	&emsp; Hap3_F: Diagonal of ROH based relationship matrix. <br> 
	  	&emsp; Homozy: Proportion of the genome homozygous. <br> 
	  	&emsp; Homolethal: Number of homozygous lethal genotypes. <br> 
	  	&emsp; Heterlethal: Number of heterozygous lethal genotypes. <br> 
	  	&emsp; Homosublethal: Number of homozygous sub-lethal genotypes. <br> 
	  	&emsp; Hetersublethal: Number of heterozygous sub-lethal genotypes. <br> 
	  	&emsp; Letequiv: Lethal equivalent value. <br> 
	  	&emsp; Fitness: Multiplicative Fitness value of the individual. <br>
	  	&emsp; GV: True genotypic value of individual (Σ (a + d)). <br>
		&emsp; BV: True breeding value of individual (Σ a). <br>
		&emsp; DD: True dominance deviation of individual (Σ d). <br>
		&emsp; QTL_Fitness: QTL genotypes of individual (0-11; 2-22; 3-12; 4-21). <br>
		</div>
	</div>
	<br>
	
	<div class = "subparameter">
		  <font color="blue">Summary Statistic Files </font>
	</div>
	<br> 
	<div class="parameter">  	
		<div class="file">	
			Summary_Statistics_DataFrame_Performance: <br>
		</div>
	  	- This provides summary statistics on the phenotypic and genetic trend across generations. <br>
	  	<div class="columnheadings">
	  	&emsp; Generation: Generation number. <br>
		&emsp; phen: Mean (variance) phenotypic value. <br>
		&emsp; ebv: Mean (variance) estimated breeding value. <br>
		&emsp; gv: Mean (variance) true genotypic breeding value (Σ (a + d)). bv: Mean (variance) true genotypic breeding value (Σ a). <br>
		&emsp; dd: Mean (variance) true dominance deviation (Σ d). <br>
		&emsp; res: Mean (variance) residual value. <br>
		</div>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Summary_Statistics_DataFrame_Inbreeding: <br>
		</div>
		- This provides summary statistics on the inbreeding and fitness trends across generations. <br>
	  	<div class="columnheadings">
	  	&emsp; Generation: Generation number. <br>
	  	&emsp; ped_f: Mean pedigree based inbreeding parameter. <br>
		&emsp; gen_f: Mean genomic relationship diagonal constructed based on Van Raden (2008). <br>
		&emsp; h1_f: Mean diagonal of haplotype based relationship matrix (Hickey et al. 2012; H1). <br>
		&emsp; h2_f: Mean diagonal of haplotype based relationship matrix (Hickey et al. 2012; H2). <br>
		&emsp; h3_f: Mean diagonal of ROH based relationship matrix (Howard et al. Submitted). <br>
		&emsp; homozy: Mean proportion homozygous (i.e. 1 - homozy = Observed Heterozygosity). <br>
		&emsp; PropROH: Mean proportion of the genome in ROH of a given length. <br>
		&emsp; ExpHet: Expected Heterozygosity (i.e. Σ (1 - p2 - q2)). <br>
		&emsp; fitness: Mean multiplicative fitness value of an individual. <br>
		&emsp; homozlethal: Mean number of homozygous FTL classified as lethal. <br>
		&emsp; hetezlethal: Mean number of heterozygous FTL classified as lethal. <br>
		&emsp; homozysublethal: Mean number of homozygous FTL classified as sub-lethal. <br>
		&emsp; hetezsublethal: Mean number of heterozygous FTL classified as sub-lethal. <br>
		&emsp; lethalequiv: Mean lethal equivalents (Lethal equivalents = Σ s for an animal). <br>
		</div>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Summary_Statistics_QTL: <br>
		</div>
		- This provides summary statistics on the number of QTL/FTL segregating across generations. <br>
	  	<div class="columnheadings">
	  	&emsp; Generation: Generation number. <br>
		&emsp; Quant Founder Start: Number of QTL from founder generation segregating. <br>
		&emsp; Quant Founder Lost: Number of QTL from founder generation fixed. <br>
		&emsp; Mutation Quan Total: Number of QTL from new mutations segregating. <br>
		&emsp; Mutation Quan Lost: Number of QTL from new mutations fixed. <br>
		&emsp; Additive Var: True additive genetic variance based on Σ 2pq[a+d(q-p)]<sup>2</sup>. <br>
		&emsp; Dominance Var: True dominance genetic variance based on Σ (2pqd)<sup>2</sup>. <br>
		&emsp; Fit Founder Start: Number of FTL from founder generation segregating. <br>
		&emsp; Fit Founder Lost: Number of FTL derived from founder generation fixed. <br>
		&emsp; Mutation Fit Total: Number of FTL derived from new mutations segregating.<br>
		&emsp; Mutation Fit Lost: Number of FTL derived from new mutations fixed. <br>
		&emsp; Avg Haplotypes Window: Mean haplotypes contained within a haplotype window. <br>
		&emsp; ProgenyDiedFitness: Number of progeny that died due to fitness. <br>
		</div>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			LD_Decay: <br>
		</div>
		- A file that has the average correlation (r<sup>2</sup>) between two SNP across a range of distances. The average was generated by moving across the genome in 10 Mb blocks 
		and randomly grabbing two SNP and calculating their respective (r<sup>2</sup>) and placing them in the correct bins based how far they were apart. Within a block 500 pairs of 
		SNP are randomly sampled and once finished the window is shifted by 5 Mb and is repeated until the end of the chromosome. This is conducted within each chromosome. The 
		distances are in the first row and are in Kilobases. Each row after the first row corresponds to the generation, such that line 2 is generation 0, line 3 is generation 1, etc. 
		The formula to calculate the D and (r2) values is below and the subscript refers to either SNP marker 1 or 2. <br>
	  	<div id = "mme">
					<p><img src="{{ site.baseurl }}/images/LD_Calc.png"></p>
		</div>
	</div>
	<div class="parameter">  	
		<div class="file">	
			QTL_LD_Decay: <br>
		</div>
		- Similar to the "LD Decay" file this file estimates the average correlation between a SNP and a QTL within window sizes of 0-0.5 Mb, 0.5-1.0 Mb, 1.0-1.5 Mb, 1.5-2.0 Mb and 2.0-2.5 Mb. 
		For example if a SNP and QTL were 0.75 Mb apart it would get placed in the 0.5 - 1.0 Mb bin. Each line in the file represents a QTL. If the correlation could not be estimated due to 
		lack of SNP in the window or if SNP are near fixation a ’-5’ will be produced. The first column represents the chromosome and the second column represents the position in Mb. The last 
		column is the average correlation within a generation across the different window sizes. Correlations within a generation across windows are separated by a ":" and sets of correlations 
		across generations are seperated by a "_". <br>
	</div>
	<br>
	</div>
	<div class="parameter">  	
		<div class="file">	
			Phase_Persistance_Generation: <br>
		</div>
		- Using the output from the Phase Persistance file, the correlation between phases across generations was estimated within window sizes of 0-0.5 Mb, 0.5-1.0 Mb, 1.0-1.5 Mb, 1.5-2.0 Mb and 
		2.0-2.5 Mb. The first column is the generation and the columns following would be the correlation between the current generation and preceding generations. <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			ProgenyParentCorrelationGeneration: <br>
		</div>
		- This provides summary statistics on training (i.e. anyone but selection candidates) and validation (i.e. selection candidates) correlations and bias. <br>
	  	<div class="columnheadings">
	  	&emsp; Generation: Generation number. <br>
		&emsp; Cor_Parent_TGV: Correlation between parents ebv and TGV (Σ(a + d)). <br>
		&emsp; Cor_Parent_TBV: Correlation between parents ebv and TBV (Σ(a)). <br>
		&emsp; Cor_Parent_TDD: Correlation between parents ebv and TDD (Σ(d)). <br>
		&emsp; Bias_Parent_TBV: Regression of TBV on parents ebv and the value is the regression coefficient. <br>
		&emsp; Cor_Progeny_TGV: Correlation between selection candidates ebv and TGV. <br>
		&emsp; Cor_Progeny_TBV: Correlation between selection candidates ebv and TBV. <br>
		&emsp; Cor_Progeny_TDD: Correlation between selection candidates ebv and TDD. <br>
		&emsp; Bias_Progeny_TBV: Regression of TBV on selection candidates ebv and the value is the regression coefficient. <br>
		</div>
	</div>
	<br>
	</div>
	<div class="parameter">  	
		<div class="file">	
			AmaxGeneration: <br>
		</div>
		- This file estimates the mean maximum relationship for individuals that were born in a previous generation with selection candidates.<br>
	</div>
	<br>
	</div>
	<div class="parameter">  	
		<div class="file">	
			Summary_Statistics_ROH_Freq: <br>
		</div>
		- The first two columns are the chromosomal and nucleotide position of the SNP and the remaining columns are the frequency of that SNP being in an ROH of the length 
		that was specified for a given generation. A SNP may not be in a window of a given length and therefore is set to -5.<br>
	</div>
	<br>
	</div>
	<div class="parameter">  	
		<div class="file">	
			Summary_Statistics_ROH_Length: <br>
		</div>
		- The first two columns are the chromosomal and nucleotide position of the SNP and the remaining columns are the mean and median length (i.e. ”Mean_Median”) of ROH 
		for that given SNP when that SNP is in a ROH for a given generation. A SNP may not be in a window of a given length and therefore is set to -5.<br>
	</div>
	<br>
	</div>
	<div class="parameter">  	
		<div class="file">	
			WindowAdditiveVariance: <br>
		</div>
		- Provides the true additive genetic variance for a given 1-Mb window across generations. The first row is the chromosome and mid-point in the window and the 
		remaining lines are the associated additive genetic variance estimates for a given window. <br>
	</div>
	<br>
	</div>
	<div class="parameter">  	
		<div class="file">	
			WindowDominanceVariance: <br>
		</div>
		- Provides the true dominance genetic variance for a given 1-Mb window across generations. The first row is the chromosome and mid-point in the window and the 
		remaining lines are the associated dominance genetic variance estimates for a given window.<br>
	</div>
	<br>
	
	
	<div class = "subparameter">
		  <font color="blue">Supplementary Files </font>
	</div>
	<br> 
	<div class="parameter">  	
		<div class="file">	
			CH*SNP.txt: <br>
		</div>
		- Haplotype sequence for each chromosome simulated from MaCS. <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			MAP*.txt: <br>
		</div>
		- Map file corresponding to haplotypes sequence in CH*SNP.txt. <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			FounderGenotypes: <br>
		</div>
		- Genotypes across chromosomes for each founder. The line number corresponds to the founder ID and the first column represents the row number of the two haplotypes that created the
		 genotype, followed by the genotype string. <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			G_Matrix: <br>
		</div>
		- The Genomic relationship matrix in binary format. <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Ginv_Matrix: <br>
		</div>
		- The inverse of the Genomic relationship matrix in binary format. <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Linv_Matrix: <br>
		</div>
		- The inverse of the cholesky matrix from the previous generation that is utilized to construct inverse relationship matrix using the method outlined by Meyer et al. (2013) to 
		obtain the inverse. This file is in binary format. <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Pheno_GMatrix: <br>
		</div>
		- Dataframe utilized in constructed genomic relationship matrix. <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			Pheno_Pedigree: <br>
		</div>
		- Used in constructing pedigree relationship matrix. <br>
	</div>
	<br>
	<div class="parameter">  	
		<div class="file">	
			SNPFreq: <br>
		</div>
		- Frequency of SNP across all chromosomes derived from MaCS. <br>
	</div>



	<div id = "homepage">
		<a href="{{site.baseurl}}/index.html">Home Page</a>
	</div>
	
	
</html>

