<!DOCTYPE html>
<html>

	<head> 
     <meta charset="utf-8">
     <link rel="stylesheet" href="{{ site.baseurl }}/css/references.css">
   	 </head> 

	<div id = "title">
	  <p>
	  		<p><img src="{{ site.baseurl }}/images/Geno_Diver_Logo_Small.png" ></p>
	  </p>
	</div>
	
	
	<div id = "parameterclass">
	 	<font color="black"> Selection and Culling Parameters </font>
	</div>	 	
		
	<div class = "main_parameter">
	  	GENERATIONS
	</div>
	<div class="parameter">  	
	  	- Description: Determines the number of generations to simulate. <br>
	  	- Option: Integer value. <br>
	  	- Usage: '<font color="darkgreen">GENERATIONS: 10</font>'. <br>
		- Type: Mandatory. <br>
		- Note:  <br>
		<div class="Note">
	  	The maximum number of generations vary depending on the size of your computer memory. At the current time if a quantitative trait is simulated the breeding values are generated by building the mixed model equations and solving them, which can take up a large amount of memory. It is advisable to start at a small number of generations and then build up to determine how much memory the scenario will use.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	INDIVIDUALS
	</div>
	<div class="parameter">  	
	  	- Description: Determines the number of males and females in each generation and replacement rate for parents each generation. Care should be taken on picking the number of offspring to have enough for the next generation. If the number of animals falls below the input male or female value, the program exits. <br>
	  	- Options:  <br>
	  	<div class="Options">
			- <font color="blue">Integer value</font>: Number of male parents within each generation. <br>
			- <font color="blue">Double value (0.0 to 1.0)</font>: Proportion of male parents that are culled and replaced by progeny each generation. <br>
			- <font color="blue">Integer value</font>: Number of female parents within each generation. <br>
			- <font color="blue">Double value (0.0 to 1.0)</font>: Proportion of female parents that are culled and replaced by progeny each generation. <br>
			</div>
	  	- Usage: '<font color="darkgreen">INDIVIDUALS: 50 0.2 600 0.2</font>'. <br>
		- Type: Mandatory. <br>
	</div>

	<div class = "main_parameter">
	  	PARITY_MATES_DIST
	</div>
	<div class="parameter">  	
	  	- Description: Determines the distribution of the number of mating pairs a sire has for each age group. A Beta distribution, which is parameterized by two parameters, is used to generate the distribution of mating pairs for a given age group. To generate the number of mating pairs by age class, the cumulative distribution function (CDF) is split into quadrants based on the number of age classes that occur within a generation. The total number of mating pairs within an age class is the proportion that falls within the CDF quadrant for a given age class. <br>
	  	- Options:  <br>
	  	<div class="Options">
			- <font color="blue">Double value (> 0.0)</font>: Alpha parameter in Beta distribution. <br>
			- <font color="blue">Double value (> 0.0)</font>: Beta parameter in Beta distribution. <br>
		</div>
	  	- Usage: '<font color="darkgreen">PARITY_MATES_DIST: 1 1</font>'. <br>
		- Type: Optional. The default is both parameters being 1, which is very similar to a uniform distribution, such that all age classes have the same proportion of mating pairs. <br>
	</div>
	
	<div class = "main_parameter">
	  	PROGENY
	</div>
	<div class="parameter">  	
	  	- Description: Determines the number of progeny produced by each mating pair. The number of progeny per mating pair may not be the actual number of progeny produced if fitness QTL segregates in the population. The progeny of a mating pair makes it to breeding age if the fitness value of the progeny is greater than a random value derived from a uniform distribution ranging from 0 to 1. The fitness value of the progeny is a function of the multiplicative effect of all lethal and sub-lethal QTL genotype effects. <br>
	  	- Option: Integer value. <br>
	  	- Usage: '<font color="darkgreen">PROGENY: 4</font>'. <br>
		- Type: Mandatory. <br>
	</div>

	<div class = "main_parameter">
	  	MAXFULLSIB
	</div>
	<div class="parameter">  	
	  	- Description: - Determines the maximum number of full-sib progeny selected within a family. Once the maximum number is reached the full-sib with the lowest selection criteria is no longer selected and the next best animal is selected. This process is repeated across all families until all families are below the value. <br>
	  	- Option: Ranges from 1 to number of progeny. <br>
	  	- Usage: '<font color="darkgreen">MAXFULLSIB: 2</font>'. <br>
		- Type: Optional. Default set at number of progeny. <br>
	</div>
	
	<div class = "main_parameter">
	  	SELECTION
	</div>
	<div class="parameter">  	
	  	- Description: The metric used to select individuals and favorable direction. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Selection Criteria: <br>
	  		&emsp; - <font color="blue">random</font>: Parent selected randomly. <br>
	  		&emsp; - <font color="blue">phenotype</font>: Parent selected based on phenotypic value. <br>
	  		&emsp; - <font color="blue">true_bv</font>: Parent selected based on true breeding value. <br>
	  		&emsp; - <font color="blue">ebv</font>: Parents selected based on estimated breeding value. <br>
	  		- Direction: <br>
	  		&emsp; - <font color="blue">high</font>: High values are favorable. <br>
	  		&emsp; - <font color="blue">low</font>: Low values are favorable. <br>
	  	</div>
	  	- Usage: '<font color="darkgreen">SELECTION: ebv high</font>'. <br>
		- Type: Mandatory. <br>
		<div class="Note">
	  	For random selection, the direction does not impact the results and therefore the direction value does not matter.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	EBV_METHOD
	</div>
	<div class="parameter">  	
	  	- Description: - Parameter that specifies how estimated breeding values are generated. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Method: <br>
	  		<div class="OptionsLongMethod">
			 - <font color="blue">pblup</font>: The best linear unbiased prediction (BLUP) of estimated breeding values are obtained by Henderson's (Henderson 1975) mixed model equations. The mixed model equations are outlined below: <br>
				<div id = "mme">
					<p><img src="{{ site.baseurl }}/images/MME.png"></p>
				</div>
				where X and Z refers to fixed and random design matrices, which relate records to fixed and random effects, A<sup>-1</sup> refers to the inverse of the pedigree based relationship matrix. Fixed effects include the intercept and random effects include the effect of the animal. The mixed model equations are solved by the preconditioned conjugate gradient method or cholesky decomposition of the LHS of the matrix. <br>
			- <font color="blue">gblup</font>: The same as the pblup option, breeding values are obtained by solving the mixed model equations, except the inverse of the pedigree based relationship matrix is replaced by the inverse of the genomic based relationship matrix (VanRaden 2008). <br>
			- <font color="blue">rohblup</font>: The same as the pblup option, breeding values are obtained by solving the mixed model equations, except the inverse of the pedigree based relationship matrix is replaced by the inverse of the run-of-homozygosity based relationship matrix (Howard et al. 2016). <br>
			- <font color="blue">bayes</font>: Marker effects are estimated utilizing bayesian whole genome regression models based on the following model:
				<div id = "mme">
					<p><img src="{{ site.baseurl }}/images/BayesModel.png"></p>
				</div>
			where y is the phenotype for individuali, μ is the intercept, SNP is the additive genetic effects that correspond to allele substitution effects for each marker and ε is the residual for individual i. SNP covariates had values of 0 for the homozygote, 1 for the heterozygote and 2 for the alternative homozygote. The intercept is assigned an uninformative prior and the SNP covariates can have 4 possible prior densities: 1.) Gaussian (BayesRidgeRegression); 2.) Scaled-t (BayesA); 3.) Two finite mixture priors: a mixture of a point of mass at zero and a Gaussian slab (BayesC); 4.) Two finite mixture priors: a mixture of a point of mass at zero and a Scaled-t slab (BayesB). Samples were drawn from the posterior density using a Gibbs sampler with scalar updating.
			</div>
		</div>
	  	- Usage: '<font color="darkgreen">EBV_METHOD: bayes</font>'. <br>
		- Type: Only mandatory if selection is based on ebv. <br>
		- Note:  <br>
		<div class="Note">
	  	If left out and selection is not based on ebv the program will not calculate breeding values. If parameter included and selection is not based on ebv, the program will calculate breeding values.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	BLUP_OPTIONS
	</div>
	<div class="parameter">  	
	  	- Description: - When ebv are estimated based on ’pblup’ or gblup’ this parameter can be included to remove older individuals from the analysis, specify solving method and how the inverse of the relationship matrix is calculated. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- <font color="blue"> Generations (integer)</font>: Number of generations to trace back from the current group of selection candidates. All animals ’n’ generations back and their associated progeny will be included in the analysis. For example, if a value of 1 is specified the selection candidate’s parents and the associated parents progeny will only be included in the analysis. The default value is the number of generations simulated (i.e. all animals are used each generation). <br>
	  		- Solver: <br>
	  		&emsp; - <font color="blue">direct</font>: Uses Cholesky decomposition for matrix inversion. When only simulating a small number of generations this is advised. <br>
	  		&emsp; - <font color="blue">pcg</font>: Uses the iterative preconditioned conjugate gradient (PCG) method and is faster than direct when the number of animals is large (Default). <br>
	  		- Genomic Inverse (only applicable if ’gblup’ is specified): <br>
	  		&emsp; - <font color="blue">cholesky</font>: Update previous inverse based on the algorithm presented by Meyer et al. (2012) (Default). <br>
	  		&emsp; - <font color="blue">recursion</font>: Utilizing the sequential update algorithm presented by Misztal et al. (2014). <br>
	  	</div>
	  	- Usage: '<font color="darkgreen">BLUP_OPTIONS: 4 pcg cholesky</font>'. <br>
		- Type: Optional. <br>
		- Note:  <br>
		<div class="Note">
	  	Parameters that specify inverse calculations are for genomicbased relationships. Calculation of pedigree-based relationships is generated based on Meuwissen & Luo (1992).
		</div>
	</div>
	
	<div class = "main_parameter">
	  	G_OPTIONS
	</div>
	<div class="parameter">  	
	  	- Description: - When ebv are estimated based on 'gblup’ this parameter can be included to determine how the genomic relationship matrix is calculated. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- How genomic relationship is constructed: <br>
	  		&emsp; - <font color="blue">VanRaden</font>: Constructed based on G=ZZ’/ 2&Sigma;(pq). <br>
	  		- Allele frequencies calculated: <br>
	  		&emsp; - <font color="blue">founder</font>: Frequencies calculated based on founder genome. <br>
	  		&emsp; - <font color="blue">current population</font>: Frequencies calculated based on animals who are parents and selection candidates. <br>
	  	</div>
	  	- Usage: '<font color="darkgreen">G_OPTIONS: VanRaden observed</font>'. <br>
		- Type: Optional. <br>
	</div>
	
	
	<div class = "main_parameter">
	  	BAYESOPTIONS
	</div>
	<div class="parameter">  	
	  	- Description: - When ebv are estimated based on 'bayes’ this parameter has to be included to determine which method to utilize and MCMC parameters. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Bayesian prediction method: (Required) <br>
	  		&emsp; - <font color="blue">BayesRidgeRegression</font>: Gaussian prior (i.e. All markers have an effect). <br>
	  		&emsp; - <font color="blue">BayesA</font>: Scaled-t prior (All markers have an effect with). <br>
	  		&emsp; - <font color="blue">BayesB</font>: Spike with a point of mass at 0 and slab with the slab having a scaled-t prior. <br>
	  		&emsp; - <font color="blue">BayesC</font>: Spike with a point of mass at 0 and slab with the slab having a gaussian prior. <br>
	  		- <font color="blue">Integer value</font>: Number of MCMC iterations the first time a method is implemented. (Required) <br>
	  		- <font color="blue">Integer value</font>: Number of MCMC iterations to burn-in the first time a method is implemented. (Required) <br>
	  		- <font color="blue">Integer value</font>: Number of MCMC iterations for any generation after the first time a method is implemented. (Required) <br>
	  		- <font color="blue">Integer value</font>: Number of MCMC iterations to burn-in for any generation after the first time a method is implemented. (Required) <br>
	  		- <font color="blue">Integer value</font>: Thinning rate for MCMC iterations (Required). <br>
	  		- How pie (i.e. proportion of markers with non-null effect) is determined: (Only needed for BayesB & BayesC) <br>
	  		&emsp; - <font color="blue">fix</font>: Fix proportion of SNP with non-null effect. <br>
	  		&emsp; - <font color="blue">estimate</font>: Estimate proportion of SNP with non-null effect. <br>
	  		- <font color="blue">Double value</font>: Proportion of markers with non-null effect. (Only needed for BayesB & BayesC) <br>
	  		- <font color="blue">Integer value</font>: Number of generations to trace back from the current group of selection candidates. All animals ’n’ generations back and their associated progeny will be included in the analysis. For example, if a value of 1 is specified the selection candidate’s parents and the associated parents progeny will only be included in the analysis. The default value is the number of generations simulated (i.e. all animals are used each generation). (Optional). <br>
	  	</div>
	  	- Usage: '<font color="darkgreen">BAYESOPTIONS: BayesC 10000 2000 6000 1000 5 fix 0.10</font>'. <br>
		- Type: If 'bayes' EBV option is utilized has to be included otherwise can't be in parameter file. <br>
		- Note:  <br>
		<div class="Note">
	  	The number of iterations to run can be investigated by looking at the 'bayes_mcmc_samples' files within the output directory. For each generation the files gets
	  	replaced by the most recent results.
		</div>
	</div>
	</div>
	
	<div class = "main_parameter">
	  	CULLING
	</div>
	<div class="parameter">  	
	  	- Description: The metric used to cull individuals and maximum age an animal can remain in the population before being removed due to old age. The direction is the same as the selection direction. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Culling Criteria: <br>
	  		&emsp; - <font color="blue">random</font>: Parent culled randomly. <br>
	  		&emsp; - <font color="blue">phenotype</font>: Parent culled based on phenotypic value. <br>
	  		&emsp; - <font color="blue">true_bv</font>: Parent culled based on true breeding value. <br>
	  		&emsp; - <font color="blue">ebv</font>: Parents culled based on estimated breeding value.<br>
	  		- Age Removed: <br>
	  		&emsp; - <font color="blue">Integer</font>: Age at which an animal has to be removed from the population. <br>
	  	 </div>
	  	- Usage: '<font color="darkgreen">CULLING: ebv 5</font>'. <br>
		- Type: Mandatory. <br>
	</div>
			
	<div id = "homepage">
		<a href="{{site.baseurl}}/index.html">Home Page</a>
	</div>
	
	
	

</html>

