<!DOCTYPE html>
<html>
	<head> 
    	 <meta charset="utf-8">
     	<link rel="stylesheet" href="{{ site.baseurl }}/css/references.css">
     	<title> GenoDiver Livestock Simulation Software </title>
     	<!-- Global site tag (gtag.js) - Google Analytics -->
		<script async src="https://www.googletagmanager.com/gtag/js?id=UA-110857544-1"></script>
		<script>
		  window.dataLayer = window.dataLayer || [];
		  function gtag(){dataLayer.push(arguments);}
		  gtag('js', new Date());

		  gtag('config', 'UA-110857544-1');
		</script>
   	</head> 
	<body>

	<div id = "header" >
   		<div id = "maintitle" >  GenoDiver </div>
   		<div id = "subtitle" > A Coalescent Forward in Time Simulation Toolkit </div>	
   		<div id = "navigation" > 
   			<a href = "index.html"> Home </a> &emsp;
   			<a href = "about.html"> About </a> &emsp;      
   			<a href = "parameters.html"> Parameters </a> &emsp;
   			<a href = "outputfiles.html"> Output Files </a> &emsp;
   			<a href = "examples.html"> Examples </a> 
   		</div>
   	</div>
	<br>
	<div id = "parameterclass">
	 	<font color="black"> Selection and Culling Parameters </font>
	</div>	 	
		
	<div class = "main_parameter">
	  	GENERATIONS
	</div>
	<div class="parameter">  	
	  	- Description: Determines the number of generations to simulate and includes the number of generations chosen with the MALE FEMALE FOUNDER option. <br>
	  	- Option: Integer value. <br>
	  	- Usage: '<font color="darkgreen">GENERATIONS: 10</font>'. <br>
		- Type: Mandatory. <br>
		- Note:  <br>
		<div class="Note">
	  	The number of generations allowed vary depending on the size of your computer memory. It is advisable to start at a small number of generations and then build up to determine how much memory a scenario will use.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	INDIVIDUALS
	</div>
	<div class="parameter">  	
	  	- Description: Determines the number of males and females in each generation and replacement rate for parents each generation. If the number of parents falls below the user specified male or female value, the program exits. <br>
	  	- Options:  <br>
	  	<div class="Options">
			- <font color="blue">Integer value</font>: Number of male parents within each generation. <br>
			- <font color="blue">Double value (0.0 to 1.0)</font>: Proportion of male parents that are culled and replaced by progeny each generation. <br>
			- <font color="blue">Integer value</font>: Number of female parents within each generation. <br>
			- <font color="blue">Double value (0.0 to 1.0)</font>: Proportion of female parents that are culled and replaced by progeny each generation. <br>
			</div>
	  	- Usage: '<font color="darkgreen">INDIVIDUALS: 50 0.2 600 0.2</font>'. <br>
		- Type: Mandatory. <br>
	</div>

	<div class = "main_parameter">
	  	PARITY_MATES_DIST
	</div>
	<div class="parameter">  	
	  	- Description: Determines the distribution of the number of mating pairs a sire has for each age group. A Beta (Alpha, Beta) is used to generate the distribution of mating pairs for a given age group. To generate the number of mating pairs by age class, the cumulative distribution function (CDF) is split into quadrants based on the number of age classes that occur within a generation. The total number of mating pairs within an age class is the proportion that falls within the CDF quadrant for a given age class. <br>
	  	- Options:  <br>
	  	<div class="Options">
			- <font color="blue">Double value (> 0.0)</font>: Alpha parameter in Beta distribution. <br>
			- <font color="blue">Double value (> 0.0)</font>: Beta parameter in Beta distribution. <br>
		</div>
	  	- Usage: '<font color="darkgreen">PARITY_MATES_DIST: 1 1</font>'. <br>
		- Type: Optional. The default is both parameters being 1, which is very similar to a uniform distribution, such that all age classes have the same proportion of mating pairs. <br>
	</div>
	
	<div class = "main_parameter">
	  	PROGENY
	</div>
	<div class="parameter">  	
	  	- Description: Determines the number of progeny produced by each mating pair. The number of progeny per mating pair may not be the actual number of progeny produced if fitness QTL segregates in the population. The progeny of a mating pair makes it to breeding age if the fitness value of the progeny is greater than a random value derived from a uniform distribution ranging from 0 to 1. The fitness value of the progeny is a function of the multiplicative effect of all lethal and sub-lethal QTL genotype effects. <br>
	  	- Option: Integer value. <br>
	  	- Usage: '<font color="darkgreen">PROGENY: 4</font>'. <br>
		- Type: Mandatory. <br>
	</div>

	<div class = "main_parameter">
	  	MAXFULLSIB
	</div>
	<div class="parameter">  	
	  	- Description: - Determines the maximum number of full-sib progeny selected within a family. Once the maximum number is reached the full-sib with the lowest selection criteria is no longer selected and the next best animal is selected. This process is repeated across all families until all families are below the value. <br>
	  	- Option: Ranges from 1 to number of progeny. <br>
	  	- Usage: '<font color="darkgreen">MAXFULLSIB: 2</font>'. <br>
		- Type: Optional. Default set at number of progeny. <br>
	</div>
	
	<div class = "main_parameter">
	  	SELECTION
	</div>
	<div class="parameter">  	
	  	- Description: The metric used to select individuals and favorable direction. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Selection Criteria: <br>
	  		&emsp; - <font color="blue">random</font>: random selection. <br>
	  		&emsp; - <font color="blue">phenotype</font>: phenotypic selection. <br>
	  		&emsp; - <font color="blue">tbv</font>: selection based on true breeding value (TBV). <br>
	  		&emsp; - <font color="blue">ebv</font>: selection based on estimated breeding value (EBV). <br>
	  		&emsp; - <font color="blue">index_tbv</font>: selection based on an index for the TBV across both traits. <br>
	  		&emsp; - <font color="blue">index_ebv</font>: selection based on an index for the EBV across both traits. <br>
	  		- Direction: <br>
	  		&emsp; - <font color="blue">high</font>: High values are favorable. <br>
	  		&emsp; - <font color="blue">low</font>: Low values are favorable. <br>
	  	</div>
	  	- Usage: '<font color="darkgreen">SELECTION: ebv high</font>'. <br>
		- Type: Mandatory. <br>
		- Note:  <br>
		<div class="Note">
	  	For random selection, the direction does not impact the results and therefore the direction option does not matter. For index tbv and index ebv two traits have to be simulated. The index for animal ‘i’ based on TBV is constructed as follows:
	  		<div id = "mme">
					<p><img src="{{ site.baseurl }}/images/IndexFig.png"></p>
			</div>
		where weight1 and weight2 are the index weights for Trait1 and Trait2, respectively. The index based on EBV is the same except EBV are used instead of TBV. The standard deviation for Trait 1 and 2 is calculated from animals born a generation before truncation selection begins.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	INDEX_PROPORTIONS
	</div>
	<div class="parameter">  	
	  	- Description: Sets the weight that each trait is given when constructing the index true or estimated breeding value when selection is based on index tbv or index ebv. <br>
	  	- Options:  <br>
	  	<div class="Options">
			- <font color="blue">Double value</font>: Weight given to Trait 1. <br>
			- <font color="blue">Double value</font>: Weight given to Trait 2. <br>
		</div>
	  	- Usage: '<font color="darkgreen">INDEX PROPORTIONS: 0.20 0.80</font>'. <br>
		- Type: Mandatory if selection based on ‘index tbv’ or ‘index ebv’. <br>
		- Note:  <br>
		<div class="Note">
	  	The index weights must be such that their sum is equal to 1 and each weight is between 0.0 and 1.0.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	EBV_METHOD
	</div>
	<div class="parameter">  	
	  	- Description: - Parameter that specifies how estimated breeding values (EBV) are generated. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Method: <br>
	  		<div class="OptionsLongMethod">
			 - <font color="blue">pblup</font>: The best linear unbiased prediction (BLUP) of EBV are obtained by Henderson’s (Henderson 1975) mixed model equations. The mixed model equations are outlined below: <br>
				<div id = "mme">
					<p><img src="{{ site.baseurl }}/images/MME.png"></p>
				</div>
				where X and Z refers to fixed and random design matrices, which relate records to fixed and random effects, A<sup>-1</sup> refers to the inverse of the pedigree based relationship matrix. Fixed effects include the intercept and random effects include the effect of the animal. The mixed model equations are solved by the preconditioned conjugate gradient method or cholesky decomposition of the LHS of the matrix. <br>
			- <font color="blue">gblup</font>: The same as the pblup option, EBV are obtained by solving the mixed model equations, except the inverse of the pedigree based relationship matrix is replaced by the inverse of the genomic based relationship matrix. <br>
			- <font color="blue">rohblup</font>: The same as the pblup option, EBV are obtained by solving the mixed model equations, except the inverse of the pedigree based relationship matrix is re- placed by the inverse of the run-of-homozygosity based relationship matrix. <br>
			- <font color="blue">ssgblup</font>: The same as the pblup option, EBV are obtained by solving the mixed model equations, except the inverse of the pedigree based relationship matrix is replaced by the inverse of a relationship matrix that augments pedigree relationships across all individuals based on contributions from animals with genomic information (H<sup>-1</sup>; Aguilar et al., 2010; Christensen & Lund, 2010). <br>
			- <font color="blue">bayes</font>: Marker effects and the associated EBV are estimated utilizing bayesian whole genome regression models based on the following model:
				<div id = "mme">
					<p><img src="{{ site.baseurl }}/images/BayesModel.png"></p>
				</div>
			where y is the phenotype for individuali, μ is the intercept, SNP is the additive genetic effects that correspond to allele substitution effects for each marker and ε is the residual for individual i. SNP covariates had values of 0 for the homozygote, 1 for the heterozygote and 2 for the alternative homozygote. The intercept is assigned an uninformative prior and the SNP covariates can have 4 possi- ble prior densities: 1.) Gaussian (BayesRidgeRegression); 2.) Scaled-t (BayesA); 3.) Two finite mixture priors: a mixture of a point of mass at zero and a Gaussian slab (BayesC); 4.) Two finite mixture priors: a mixture of a point of mass at zero and a Scaled-t slab (BayesB). Sam- ples were drawn from the posterior density using a Gibbs sampler with scalar updating.
			</div>
		</div>
	  	- Usage: '<font color="darkgreen">EBV_METHOD: pblup</font>'. <br>
		- Type: Only mandatory if selection is based on ebv. <br>
		- Note:  <br>
		<div class="Note">
	  	If left out and selection is not based on ebv the program will not calculate breeding values. If parameter included and selection is not based on ebv, the program will calculate breeding values.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	BLUP_OPTIONS
	</div>
	<div class="parameter">  	
	  	- Description: - When EBV are estimated based on ’pblup’, gblup’, ’rohblup’ or ’ssgblup’ this parameter can be included to remove older individuals from the analysis, specify solving method and how the inverse of the relationship matrix is calculated. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- <font color="blue"> Generations (integer)</font>: Number of generations to trace back from the current group of selection candidates. All animals ’n’ generations back and all of the progeny of the associated parents will be included in the analysis. For example, if a value of 1 is specified the selection candidate’s parents and all progeny that were parents of the selection candidates parents will only be included in the analysis. The default value is the number of generations simulated (i.e. all animals are used each generation). <br>
	  		- Solver: <br>
	  		&emsp; - <font color="blue">direct</font>: Uses Cholesky decomposition for matrix inversion. If EBV accuracy is needed or only simulating a small number of generations this is advised. <br>
	  		&emsp; - <font color="blue">pcg</font>: Uses the iterative preconditioned conjugate gradient (PCG) method and is faster than direct when the number of animals is large. The EBV accuracy is not calculated (Default). <br>
	  		- Genomic Inverse (only applicable if ’gblup’ is specified): <br>
	  		&emsp; - <font color="blue">cholesky</font>: Update previous Cholesky inverse (Meyer et al. 2012; Default). <br>
	  		&emsp; - <font color="blue">recursion</font>: Sequential update (Misztal et al. 2014).
	  	</div>
	  	- Usage: '<font color="darkgreen">BLUP_OPTIONS: 4 pcg cholesky</font>'. <br>
		- Type: Optional. <br>
		- Note:  <br>
		<div class="Note">
	  	Parameters that specify inverse calculations are for genomic-based relationships. Calculation of pedigree-based relationships is generated based on Meuwissen & Luo (1992).
		</div>
	</div>
	
	<div class = "main_parameter">
	  	G_OPTIONS
	</div>
	<div class="parameter">  	
	  	- Description: - When ebv are estimated based on 'gblup’ this parameter can be included to determine how the genomic relationship matrix is calculated. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- How genomic relationship is constructed: <br>
	  		&emsp; - <font color="blue">VanRaden</font>: Constructed based on G=ZZ’/ 2&Sigma;(pq). <br>
	  		- Allele frequencies calculated: <br>
	  		&emsp; - <font color="blue">founder</font>: Frequencies calculated based on founder genome. <br>
	  		&emsp; - <font color="blue">observed</font>: Frequencies calculated based on animals who are selection candidates. <br>
	  	</div>
	  	- Usage: '<font color="darkgreen">G_OPTIONS: VanRaden observed</font>'. <br>
		- Type: Optional. <br>
	</div>
	
	<div class = "main_parameter">
	  	BLENDING_GA22
	</div>
	<div class="parameter">  	
	  	- Description: - When using the ‘ssgblup’ option, a weighted genomic relationship matrix (Gw), as proposed by VanRaden (2008), is utilized when constructing the H matrix. Gw is constructed as Gw = weight1 * G + weight * A22, where G is the raw genomic relationship matrix and A22 is the pedigree-based relationship for genotyped animals.<br>
	  	- Options:  <br>
	  	<div class="Options">
	  		&emsp; - <font color="blue">Double value</font>: Weight given to the G. <br>
	  		&emsp; - <font color="blue">Double value</font>: Weight given to the A22. <br>
	  	</div>
	  	- Usage: '<font color="darkgreen">BLENDING GA22: 0.95 0.05</font>'. <br>
		- Type: Optional. Default values are 0.95 for G and 0.05 for A22. <br>
	</div>
	
	
	<div class = "main_parameter">
	  	BAYESOPTIONS
	</div>
	<div class="parameter">  	
	  	- Description: - When ebv are estimated based on the “bayes” option, this parameter has to be included to determine which method to utilize and MCMC options. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Bayesian prediction method: (Required) <br>
	  		&emsp; - <font color="blue">BayesRidgeRegression</font>: Gaussian prior. <br>
	  		&emsp; - <font color="blue">BayesA</font>: Scaled-t prior. <br>
	  		&emsp; - <font color="blue">BayesB</font>: Spike with a point of mass at 0 and slab with the slab having a scaled-t prior. <br>
	  		&emsp; - <font color="blue">BayesC</font>: Spike with a point of mass at 0 and slab with the slab having a gaussian prior. <br>
	  		- <font color="blue">Integer value</font>: Number of MCMC iterations the first time a method is implemented. (Required) <br>
	  		- <font color="blue">Integer value</font>: Number of MCMC iterations to burn-in the first time a method is implemented. (Required) <br>
	  		- <font color="blue">Integer value</font>: Number of MCMC iterations for any generation after the first time a method is implemented. (Required) <br>
	  		- <font color="blue">Integer value</font>: Number of MCMC iterations to burn-in for any generation after the first time a method is implemented. (Required) <br>
	  		- <font color="blue">Integer value</font>: Thinning rate for MCMC iterations (Required). <br>
	  		- How pie (i.e. proportion of markers with non-null effect) is determined: (Only needed for BayesB & BayesC) <br>
	  		&emsp; - <font color="blue">fix</font>: Fix proportion of SNP with non-null effect. <br>
	  		&emsp; - <font color="blue">estimate</font>: Estimate proportion of SNP with non-null effect. <br>
	  		- <font color="blue">Double value</font>: Proportion of markers with non-null effect. (Only needed for BayesB & BayesC) <br>
	  		- <font color="blue">Integer value</font>: Number of generations to trace back from the current group of selection candidates. All animals ’n’ generations back and their associated progeny will be included in the analysis. For example, if a value of 1 is specified the selection candidate’s parents and the associated parents progeny will only be included in the analysis. The default value is the number of generations simulated (i.e. all animals are used each generation). (Optional). <br>
	  	</div>
	  	- Usage: '<font color="darkgreen">BAYESOPTIONS: BayesC 10000 2000 6000 1000 5 fix 0.10</font>'. <br>
		- Type: If 'bayes' EBV option is utilized has to be included otherwise can't be in parameter file. <br>
		- Note:  <br>
		<div class="Note">
	  	The number of iterations to run can be investigated by looking at the 'bayes_mcmc_samples' files within the output directory. For each generation the files gets
	  	replaced by the most recent results.
		</div>
	</div>
	</div>
	
	<div class = "main_parameter">
	  	CULLING
	</div>
	<div class="parameter">  	
	  	- Description: The metric used to cull parents and maximum age an animal can remain in the population before being removed due to old age. The direction is the same as the selection direction. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Culling Criteria: <br>
	  		&emsp; - <font color="blue">random</font>: random culling. <br>
	  		&emsp; - <font color="blue">phenotype</font>: phenotypic culling. <br>
	  		&emsp; - <font color="blue">tbv</font>: culling based on true breeding value (TBV). <br>
	  		&emsp; - <font color="blue">ebv</font>: culling based on estimated breeding value (EBV). <br>
	  		&emsp; - <font color="blue">index_tbv</font>: culling based on an index for the TBV across both traits. <br>
	  		&emsp; - <font color="blue">index_ebv</font>: culling based on an index for the EBV across both traits. <br>
	  		- Age Removed: <br>
	  		&emsp; - <font color="blue">Integer</font>: Age at which an animal has to be removed from the population. <br>
	  	 </div>
	  	- Usage: '<font color="darkgreen">CULLING: ebv 5</font>'. <br>
		- Type: Mandatory. <br>
	</div>
	
	<div class = "main_parameter">
	  	INTERIM_EBV
	</div>
	<div class="parameter">  	
	  	- Description: Determines whether to calculate interim estimated breeding values (EBV) before or after culling of parents. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		&emsp; - <font color="blue">no</font>: Interim EBV are not calculated. <br>
	  		&emsp; - <font color="blue">before_culling</font>: Interim EBV calculated prior to culling parents. <br>
	  		&emsp; - <font color="blue">after_culling</font>: Interim EBV calculated after culling parents. <br>
	  	 </div>
	  	- Usage: '<font color="darkgreen">INTERIM EBV: no</font>'. <br>
		- Type: Optional. Default is ‘no’. <br>
	</div>
	
	<div class = "main_parameter">
	  	PHENOTYPE_STRATEGY
	</div>
	<div class="parameter">  	
	  	- Description: - When simulating 1 quantitative trait, determines when and if an animal has a phenotype when estimated breeding value (EBV) are being predicted. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Male Proportion Phenotyped: <br>
	  		&emsp; - <font color="blue">Double value (0.0-1.0)</font>: Proportion of males that are phenotyped based on the male phenotyping strategy. <br>
	  		- Male Phenotyping Strategy: <br>
	  		&emsp; - <font color="blue"> pheno_atselection</font>: Phenotype selection candidates prior to predicting EBV.<br>
			&emsp; - <font color="blue"> pheno_afterselection</font>: Phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> pheno_parents</font>: Only phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> random_atselection</font>: Randomly phenotype selection candidates prior to predicting EBV.<br>
			&emsp; - <font color="blue"> random_afterselection</font>: Randomly phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> random_parents</font>: Randomly phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> ebv_atselection</font>: Phenotype selection candidates based on EBV prior to predicting EBV.<br>
			&emsp; - <font color="blue"> ebv_afterselection</font>: Phenotype selection candidates based on EBV after predicting EBV.<br>
			&emsp; - <font color="blue"> ebv_parents</font>: Phenotype selection candidates that were selected based on EBV to serve as parents.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter prior to predicting EBV.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter after predicting EBV.<br>
	  		- Female Proportion Phenotyped: <br>
	  		&emsp; - <font color="blue">Double value (0.0-1.0)</font>: Proportion of females that are phenotyped based on the female phenotyping strategy. <br>
	  		- Female Phenotyping Strategy: <br>
	  		&emsp; - <font color="blue"> pheno_atselection</font>: Phenotype selection candidates prior to predicting EBV.<br>
	  		&emsp; - <font color="blue"> pheno_afterselection</font>: Phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> pheno_parents</font>: Only phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> random_atselection</font>: Randomly phenotype selection candidates prior to predicting EBV.<br>
			&emsp; - <font color="blue"> random_afterselection</font>: Randomly phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> random_parents</font>: Randomly phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> ebv_atselection</font>: Phenotype selection candidates based on EBV prior to predicting breeding values.<br>
			&emsp; - <font color="blue"> ebv_afterselection</font>: Phenotype selection candidates based on EBV after predicting breeding values.<br>
			&emsp; - <font color="blue"> ebv_parents</font>: Phenotype selection candidates that were selected based on EBV to serve as parents.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter prior to predicting EBV.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter after predicting EBV.<br>
	  		- Distribution sampling from when doing EBV based phenotype strategy: <br>
	  		&emsp; - <font color="blue"> high</font>: Phenotype animals with a high EBV. <br>
			&emsp; - <font color="blue"> low</font>: Phenotype animals with a low EBV. <br>
			&emsp; - <font color="blue"> tails</font>: Phenotype animals with a high and low EBV.<br>
	  	</div>
	  	- Usage: '<font color="darkgreen">PHENOTYPE_STRATEGY: 1.0 atselection 1.0 atselection</font>'. <br>
		- Type: Optional. The default setting is all male and female selection candidates are phenotyped prior to selection. <br>
		- Note:  <br>
		<div class="Note">
	  	For the pheno_* the proportion has to be 0.0 or 1.0, while any other scenario has to be greater than 0.0 and less than 1.0. Both male and female proportions can’t be 0.0.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	PHENOTYPE_STRATEGY1
	</div>
	<div class="parameter">  	
	  	- Description: - When simulating 2 quantitative traits, determines when and if an animal has a phenotype when estimated breeding value (EBV) are being predicted for trait 1. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Male Proportion Phenotyped: <br>
	  		&emsp; - <font color="blue">Double value (0.0-1.0)</font>: Proportion of males that are phenotyped based on the male phenotyping strategy. <br>
	  		- Male Phenotyping Strategy: <br>
	  		&emsp; - <font color="blue"> pheno_atselection</font>: Phenotype selection candidates prior to predicting EBV.<br>
			&emsp; - <font color="blue"> pheno_afterselection</font>: Phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> pheno_parents</font>: Only phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> random_atselection</font>: Randomly phenotype selection candidates prior to predicting EBV.<br>
			&emsp; - <font color="blue"> random_afterselection</font>: Randomly phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> random_parents</font>: Randomly phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> ebv_atselection</font>: Phenotype selection candidates based on EBV prior to predicting EBV.<br>
			&emsp; - <font color="blue"> ebv_afterselection</font>: Phenotype selection candidates based on EBV after predicting EBV.<br>
			&emsp; - <font color="blue"> ebv_parents</font>: Phenotype selection candidates that were selected based on EBV to serve as parents.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter prior to predicting EBV.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter after predicting EBV.<br>
	  		- Female Proportion Phenotyped: <br>
	  		&emsp; - <font color="blue">Double value (0.0-1.0)</font>: Proportion of females that are phenotyped based on the female phenotyping strategy. <br>
	  		- Female Phenotyping Strategy: <br>
	  		&emsp; - <font color="blue"> pheno_atselection</font>: Phenotype selection candidates prior to predicting EBV.<br>
	  		&emsp; - <font color="blue"> pheno_afterselection</font>: Phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> pheno_parents</font>: Only phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> random_atselection</font>: Randomly phenotype selection candidates prior to predicting EBV.<br>
			&emsp; - <font color="blue"> random_afterselection</font>: Randomly phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> random_parents</font>: Randomly phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> ebv_atselection</font>: Phenotype selection candidates based on EBV prior to predicting breeding values.<br>
			&emsp; - <font color="blue"> ebv_afterselection</font>: Phenotype selection candidates based on EBV after predicting breeding values.<br>
			&emsp; - <font color="blue"> ebv_parents</font>: Phenotype selection candidates that were selected based on EBV to serve as parents.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter prior to predicting EBV.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter after predicting EBV.<br>
	  		- Distribution sampling from when doing EBV based phenotype strategy: <br>
	  		&emsp; - <font color="blue"> high</font>: Phenotype animals with a high EBV. <br>
			&emsp; - <font color="blue"> low</font>: Phenotype animals with a low EBV. <br>
			&emsp; - <font color="blue"> tails</font>: Phenotype animals with a high and low EBV.<br>
	  	</div>
	  	- Usage: '<font color="darkgreen">PHENOTYPE_STRATEGY1: 1.0 atselection 1.0 atselection</font>'. <br>
		- Type: Optional. The default setting is all male and female selection candidates are phenotyped prior to predicting breeding values for the first trait. <br>
		- Note:  <br>
		<div class="Note">
	  	For the pheno_* the proportion has to be 0.0 or 1.0, while any other scenario has to be greater than 0.0 and less than 1.0. Both male and female proportions can’t be 0.0.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	PHENOTYPE_STRATEGY2
	</div>
	<div class="parameter">  	
	  	- Description: - When simulating 2 quantitative traits, determines when and if an animal has a phenotype when estimated breeding value (EBV) are being predicted for trait 2. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Male Proportion Phenotyped: <br>
	  		&emsp; - <font color="blue">Double value (0.0-1.0)</font>: Proportion of males that are phenotyped based on the male phenotyping strategy. <br>
	  		- Male Phenotyping Strategy: <br>
	  		&emsp; - <font color="blue"> pheno_atselection</font>: Phenotype selection candidates prior to predicting EBV.<br>
			&emsp; - <font color="blue"> pheno_afterselection</font>: Phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> pheno_parents</font>: Only phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> random_atselection</font>: Randomly phenotype selection candidates prior to predicting EBV.<br>
			&emsp; - <font color="blue"> random_afterselection</font>: Randomly phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> random_parents</font>: Randomly phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> ebv_atselection</font>: Phenotype selection candidates based on EBV prior to predicting EBV.<br>
			&emsp; - <font color="blue"> ebv_afterselection</font>: Phenotype selection candidates based on EBV after predicting EBV.<br>
			&emsp; - <font color="blue"> ebv_parents</font>: Phenotype selection candidates that were selected based on EBV to serve as parents.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter prior to predicting EBV.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter after predicting EBV.<br>
	  		- Female Proportion Phenotyped: <br>
	  		&emsp; - <font color="blue">Double value (0.0-1.0)</font>: Proportion of females that are phenotyped based on the female phenotyping strategy. <br>
	  		- Female Phenotyping Strategy: <br>
	  		&emsp; - <font color="blue"> pheno_atselection</font>: Phenotype selection candidates prior to predicting EBV.<br>
	  		&emsp; - <font color="blue"> pheno_afterselection</font>: Phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> pheno_parents</font>: Only phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> random_atselection</font>: Randomly phenotype selection candidates prior to predicting EBV.<br>
			&emsp; - <font color="blue"> random_afterselection</font>: Randomly phenotype selection candidates after predicting EBV.<br>
			&emsp; - <font color="blue"> random_parents</font>: Randomly phenotype selection candidates that were selected to serve as parents.<br>
			&emsp; - <font color="blue"> ebv_atselection</font>: Phenotype selection candidates based on EBV prior to predicting breeding values.<br>
			&emsp; - <font color="blue"> ebv_afterselection</font>: Phenotype selection candidates based on EBV after predicting breeding values.<br>
			&emsp; - <font color="blue"> ebv_parents</font>: Phenotype selection candidates that were selected based on EBV to serve as parents.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter prior to predicting EBV.<br>
			&emsp; - <font color="blue"> litterrandom_atselection</font>: Randomly phenotype within each litter after predicting EBV.<br>
	  		- Distribution sampling from when doing EBV based phenotype strategy: <br>
	  		&emsp; - <font color="blue"> high</font>: Phenotype animals with a high EBV. <br>
			&emsp; - <font color="blue"> low</font>: Phenotype animals with a low EBV. <br>
			&emsp; - <font color="blue"> tails</font>: Phenotype animals with a high and low EBV.<br>
	  	</div>
	  	- Usage: '<font color="darkgreen">PHENOTYPE_STRATEGY2: 1.0 atselection 1.0 atselection</font>'. <br>
		- Type: Optional. The default setting is all male and female selection candidates are phenotyped prior to predicting breeding values for the second trait. <br>
		- Note:  <br>
		<div class="Note">
	  	For the pheno_* the proportion has to be 0.0 or 1.0, while any other scenario has to be greater than 0.0 and less than 1.0. Both male and female proportions can’t be 0.0.
		</div>
	</div>
	
	<div class = "main_parameter">
	  	GENOTYPE_STRATEGY
	</div>
	<div class="parameter">  	
	  	- Description: - If doing ‘ssgblup’ based BLUP determines who and at what time an animal is genotyped. <br>
	  	- Options:  <br>
	  	<div class="Options">
	  		- Generation (<font color="blue">integer value</font>): Generation to begin genotyping. <br>
	  		- Male Proportion Genotyped: <br>
	  		&emsp; - <font color="blue">Double value (0.0-1.0)</font>: Proportion of males that are genotyped based on the male genotyping strategy. <br>
	  		- Male Genotyping Strategy: <br>
	  		&emsp; - <font color="blue"> parents</font>: Genotype all parents. <br>
			&emsp; - <font color="blue"> offspring</font>: Genotype all selection candidates. <br>
			&emsp; - <font color="blue"> parents_offspring</font>: Genotype all parents and selection candidates. <br>
			&emsp; - <font color="blue"> random</font>: Randomly genotype a proportion of the selection candidates. <br>
			&emsp; - <font color="blue"> parents_random</font>: Randomly genotype a proportion of the selection candidates and all selection candidates that were selected to become parents. <br>
			&emsp; - <font color="blue"> ebv</font>: Genotype a proportion of the selection candidates based on EBV. <br>
			&emsp; - <font color="blue"> parents_ebv</font>: Genotype a proportion of the selection candidates based on EBV and all selection candidates that were selected to become parents. <br>
			&emsp; - <font color="blue"> litter_random</font>: Genotype a random proportion of the animals within each litter. <br>
			&emsp; - <font color="blue"> litter_parents_random</font>: Genotype a random proportion of the animals within each litter and all selection candidates that were selected to become parents. <br>	  	
	  		- Female Proportion Genotyped: <br>
	  		&emsp; - <font color="blue">Double value (0.0-1.0)</font>: Proportion of females that are genotyped based on the female genotyping strategy. <br>
	  		- Female Genotyping Strategy: <br>
	  		&emsp; - <font color="blue"> parents</font>: Genotype all parents. <br>
			&emsp; - <font color="blue"> offspring</font>: Genotype all selection candidates. <br>
			&emsp; - <font color="blue"> parents_offspring</font>: Genotype all parents and selection candidates. <br>
			&emsp; - <font color="blue"> random</font>: Randomly genotype a proportion of the selection candidates. <br>
			&emsp; - <font color="blue"> parents_random</font>: Randomly genotype a proportion of the selection candidates and all selection candidates that were selected to become parents. <br>
			&emsp; - <font color="blue"> ebv</font>: Genotype a proportion of the selection candidates based on EBV. <br>
			&emsp; - <font color="blue"> parents_ebv</font>: Genotype a proportion of the selection candidates based on EBV and all selection candidates that were selected to become parents. <br>
			&emsp; - <font color="blue"> litter_random</font>: Genotype a random proportion of the animals within each litter. <br>
			&emsp; - <font color="blue"> litter_parents_random</font>: Genotype a random proportion of the animals within each litter and all selection candidates that were selected to become parents. <br>	  	
	  		- Distribution sampling from when doing EBV based genotyping strategy:
	  		&emsp; - <font color="blue"> high</font>: Genotype animals with a high EBV. <br>
			&emsp; - <font color="blue"> low</font>: Genotype animals with a low EBV. <br>
			&emsp; - <font color="blue"> tails</font>: Genotype animals with a high and low EBV.<br>
	  	</div>
	  	- Usage: '<font color="darkgreen">GENOTYPE_STRATEGY: 1.0 parents 1.0 parents</font>'. <br>
		- Type: Mandatory if doing ‘ssgblup’ option. <br>
		- Note:  <br>
		<div class="Note">
	  	For the parents, offspring and parents_offspring option the proportion has to be 0.0 or 1.0, while any other scenario has to be greater than 0.0 and less than 1.0. The distribution to sample is not required and if not provided it will be the direction animals are selected on.
		</div>
	</div>
	
	<div id ="genodiverimage">
	  <p> <img src="{{ site.baseurl }}/images/GenoDiver.png" > </p>
	</div>
	
	</body>
</html>

